<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />

  <title>scanpy.external.tl.palantir &mdash; Scanpy 1.8.2 documentation</title>

<link rel=stylesheet href="https://cdn.jsdelivr.net/npm/docsearch.js@2.6/dist/cdn/docsearch.min.css"/>

      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/scanpy.css" type="text/css" />
      <link rel="stylesheet" href="../_static/plot_directive.css" type="text/css" />
      <link rel="stylesheet" href="../_static/typehints.css" type="text/css" />
      <link rel="stylesheet" href="https://assets.readthedocs.com/css/badge_only.css" type="text/css" />
    <link rel="canonical" href="https://scanpy.readthedocs.io/en/stable/generated/scanpy.external.tl.palantir.html" />

  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script async="async" src="https://assets.readthedocs.com/javascript/readthedocs-doc-embed.js"></script>
    <script src="../_static/js/theme.js"></script>

<script src="https://cdn.jsdelivr.net/npm/docsearch.js@2.6/dist/cdn/docsearch.min.js"></script>
<script>
  document.addEventListener('DOMContentLoaded', () => docsearch({
    apiKey: 'fa4304eb95d2134997e3729553a674b2',
    indexName: 'scanpy',
    inputSelector: '#rtd-search-form > input[name="q"]',
    algoliaOptions: { facetFilters: ['version:latest'] },
    debug: false,
  }))
</script>


    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="scanpy.external.tl.trimap" href="scanpy.external.tl.trimap.html" />
    <link rel="prev" title="scanpy.external.tl.phate" href="scanpy.external.tl.phate.html" />
 



<!-- RTD Extra Head -->

<link rel="stylesheet" href="https://assets.readthedocs.com/css/readthedocs-doc-embed.css" type="text/css" />

<script type="application/json" id="READTHEDOCS_DATA">{"ad_free": false, "api_host": "https://readthedocs.com", "build_date": "2021-11-03T17:53:53Z", "builder": "sphinx", "canonical_url": null, "commit": "8736fc3b", "docroot": "/docs/", "features": {"docsearch_disabled": false}, "global_analytics_code": "UA-17997319-2", "language": "en", "page": "generated/scanpy.external.tl.palantir", "programming_language": "py", "project": "icb-scanpy", "proxied_api_host": "/_", "source_suffix": ".rst", "subprojects": {}, "theme": "scanpydoc", "user_analytics_code": "", "version": "stable"}</script>

<!--
Using this variable directly instead of using `JSON.parse` is deprecated.
The READTHEDOCS_DATA global variable will be removed in the future.
-->
<script type="text/javascript">
READTHEDOCS_DATA = JSON.parse(document.getElementById('READTHEDOCS_DATA').innerHTML);
</script>

<script type="text/javascript" src="https://assets.readthedocs.com/javascript/readthedocs-analytics.js" async="async"></script>

<!-- end RTD <extrahead> -->
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html">
            <img src="../_static/Scanpy_Logo_BrightFG.svg" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                stable
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../usage-principles.html">Usage Principles</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api.html">API</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../external.html">External API</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../external.html#preprocessing-pp">Preprocessing: PP</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../external.html#tools-tl">Tools: TL</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="../external.html#embeddings">Embeddings</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.phate.html">scanpy.external.tl.phate</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="#">scanpy.external.tl.palantir</a></li>
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.trimap.html">scanpy.external.tl.trimap</a></li>
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.sam.html">scanpy.external.tl.sam</a></li>
</ul>
</li>
<li class="toctree-l3 current"><a class="reference internal" href="../external.html#clustering-and-trajectory-inference">Clustering and trajectory inference</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.phenograph.html">scanpy.external.tl.phenograph</a></li>
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.harmony_timeseries.html">scanpy.external.tl.harmony_timeseries</a></li>
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.wishbone.html">scanpy.external.tl.wishbone</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="#">scanpy.external.tl.palantir</a></li>
<li class="toctree-l4"><a class="reference internal" href="scanpy.external.tl.palantir_results.html">scanpy.external.tl.palantir_results</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../external.html#gene-scores-cell-cycle">Gene scores, Cell cycle</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../external.html#plotting-pl">Plotting: PL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../external.html#exporting">Exporting</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../ecosystem.html">Ecosystem</a></li>
<li class="toctree-l1"><a class="reference internal" href="../release-notes/index.html">Release notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../community.html">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../news.html">News</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dev/index.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Scanpy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
          <li><a href="../external.html">External API</a> &raquo;</li>
      <li>scanpy.external.tl.palantir</li>
      <li class="wy-breadcrumbs-aside">
              <!-- User defined GitHub URL -->
              <a href="https://github.com/theislab/scanpy/tree/8736fc3b02a6af3b7214481c2cf78a3f8f5e06bd/scanpy/external/tl/_palantir.py#L12-L237" class="fa fa-github"> Edit on GitHub</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="scanpy-external-tl-palantir">
<h1>scanpy.external.tl.palantir<a class="headerlink" href="#scanpy-external-tl-palantir" title="Permalink to this headline">ÔÉÅ</a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="scanpy.external.tl.palantir">
<span class="sig-prename descclassname"><span class="pre">scanpy.external.tl.</span></span><span class="sig-name descname"><span class="pre">palantir</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_components</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">knn</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_adjacency_matrix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">distances_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_eigs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">impute_data</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_steps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">copy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#scanpy.external.tl.palantir" title="Permalink to this definition">ÔÉÅ</a></dt>
<dd><p>Run Diffusion maps using the adaptive anisotropic kernel <a class="reference internal" href="../references.html#setty18" id="id1"><span>[Setty18]</span></a>.</p>
<p>Palantir is an algorithm to align cells along differentiation trajectories.
Palantir models differentiation as a stochastic process where stem cells
differentiate to terminally differentiated cells by a series of steps through
a low dimensional phenotypic manifold. Palantir effectively captures the
continuity in cell states and the stochasticity in cell fate determination.
Palantir has been designed to work with multidimensional single cell data
from diverse technologies such as Mass cytometry and single cell RNA-seq.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>More information and bug reports <a class="reference external" href="https://github.com/dpeerlab/Palantir">here</a>.</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>adata</strong> : <span class="annotation terse"><a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.html#anndata.AnnData" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code></a></span><span class="annotation full"><a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.html#anndata.AnnData" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code></a></span></dt><dd><p>An AnnData object.</p>
</dd>
<dt><strong>n_components</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">10</span></code>)</dt><dd><p>Number of diffusion components.</p>
</dd>
<dt><strong>knn</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">30</span></code>)</dt><dd><p>Number of nearest neighbors for graph construction.</p>
</dd>
<dt><strong>alpha</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">0</span></code>)</dt><dd><p>Normalization parameter for the diffusion operator.</p>
</dd>
<dt><strong>use_adjacency_matrix</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">False</span></code>)</dt><dd><p>Use adaptive anisotropic adjacency matrix, instead of PCA projections
(default) to compute diffusion components.</p>
</dd>
<dt><strong>distances_key</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code></a> | <a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.10)"><code class="xref py py-obj docutils literal notranslate"><span class="pre">None</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code></a>[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code></a>]</span> (default: <code class="docutils literal notranslate"><span class="pre">None</span></code>)</dt><dd><p>With <code class="docutils literal notranslate"><span class="pre">use_adjacency_matrix=True</span></code>, use the indicated distances key for <code class="docutils literal notranslate"><span class="pre">.obsp</span></code>.
If <code class="docutils literal notranslate"><span class="pre">None</span></code>, <code class="docutils literal notranslate"><span class="pre">'distances'</span></code>.</p>
</dd>
<dt><strong>n_eigs</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a> | <a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.10)"><code class="xref py py-obj docutils literal notranslate"><span class="pre">None</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code></a>[<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a>]</span> (default: <code class="docutils literal notranslate"><span class="pre">None</span></code>)</dt><dd><p>Number of eigen vectors to use. If <code class="docutils literal notranslate"><span class="pre">None</span></code> specified, the number of eigen
vectors will be determined using eigen gap. Passed to
<code class="docutils literal notranslate"><span class="pre">palantir.utils.determine_multiscale_space</span></code>.</p>
</dd>
<dt><strong>impute_data</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">True</span></code>)</dt><dd><p>Impute data using MAGIC.</p>
</dd>
<dt><strong>n_steps</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">3</span></code>)</dt><dd><p>Number of steps in the diffusion operator. Passed to
<code class="docutils literal notranslate"><span class="pre">palantir.utils.run_magic_imputation</span></code>.</p>
</dd>
<dt><strong>copy</strong> : <span class="annotation terse"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></a></span> (default: <code class="docutils literal notranslate"><span class="pre">False</span></code>)</dt><dd><p>Return a copy instead of writing to <code class="docutils literal notranslate"><span class="pre">adata</span></code>.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p><span class="annotation terse"><a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.html#anndata.AnnData" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code></a> | <a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.10)"><code class="xref py py-obj docutils literal notranslate"><span class="pre">None</span></code></a></span><span class="annotation full"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><code class="xref py py-data docutils literal notranslate"><span class="pre">Optional</span></code></a>[<a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.html#anndata.AnnData" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code></a>]</span></p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p><p>Depending on <code class="docutils literal notranslate"><span class="pre">copy</span></code>, returns or updates <code class="docutils literal notranslate"><span class="pre">adata</span></code> with the following fields:</p>
<dl class="simple">
<dt><strong>Diffusion maps</strong>,</dt><dd><p>used for magic imputation, and to generate multi-scale data matrix,</p>
<ul class="simple">
<li><dl class="simple">
<dt>X_palantir_diff_comp - <a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.21)"><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></a> (<a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.obsm.html#anndata.AnnData.obsm" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">obsm</span></code></a>, dtype <code class="docutils literal notranslate"><span class="pre">float</span></code>)</dt><dd><p>Array of Diffusion components.</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>palantir_EigenValues - <a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.21)"><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></a> (<a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.uns.html#anndata.AnnData.uns" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">uns</span></code></a>, dtype <code class="docutils literal notranslate"><span class="pre">float</span></code>)</dt><dd><p>Array of corresponding eigen values.</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>palantir_diff_op - <a class="reference external" href="https://docs.scipy.org/doc/scipy/reference/reference/generated/scipy.sparse.spmatrix.html#scipy.sparse.spmatrix" title="(in SciPy v1.7.1)"><code class="xref py py-class docutils literal notranslate"><span class="pre">spmatrix</span></code></a> (<a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.obsp.html#anndata.AnnData.obsp" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">obsp</span></code></a>, dtype <code class="docutils literal notranslate"><span class="pre">float</span></code>)</dt><dd><p>The diffusion operator matrix.</p>
</dd>
</dl>
</li>
</ul>
</dd>
<dt><strong>Multi scale space results</strong>,</dt><dd><p>used to build tsne on diffusion components, and to compute branch probabilities
and waypoints,</p>
<ul class="simple">
<li><dl class="simple">
<dt>X_palantir_multiscale - <a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.21)"><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></a> (<a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.obsm.html#anndata.AnnData.obsm" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">obsm</span></code></a>, dtype <code class="docutils literal notranslate"><span class="pre">float</span></code>)</dt><dd><p>Multi scale data matrix.</p>
</dd>
</dl>
</li>
</ul>
</dd>
<dt><strong>MAGIC imputation</strong>,</dt><dd><p>used for plotting gene expression on tsne, and gene expression trends,</p>
<ul class="simple">
<li><dl class="simple">
<dt>palantir_imp - <a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.21)"><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></a> (<a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.layers.html#anndata.AnnData.layers" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">layers</span></code></a>, dtype <code class="docutils literal notranslate"><span class="pre">float</span></code>)</dt><dd><p>Imputed data matrix (MAGIC imputation).</p>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</p>
</dd>
</dl>
<p class="rubric">Example</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">scanpy.external</span> <span class="k">as</span> <span class="nn">sce</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
</pre></div>
</div>
<p>A sample data is available <a class="reference external" href="https://github.com/dpeerlab/Palantir/tree/master/data">here</a>.</p>
<p><strong>Load sample data</strong></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s2">&quot;Palantir/data/marrow_sample_scseq_counts.csv.gz&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Cleanup and normalize</em></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_counts</span><span class="o">=</span><span class="mi">1000</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_counts</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_per_cell</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Data preprocessing</strong></p>
<p>Palantir builds diffusion maps using one of two optional inputs:</p>
<p><em>Principal component analysis</em></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">n_comps</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
</pre></div>
</div>
<p>or,</p>
<p><em>Nearist neighbors graph</em></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">neighbors</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Diffusion maps</em></p>
<p>Palantir determines the diffusion maps of the data as an estimate of the low
dimensional phenotypic manifold of the data.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sce</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">palantir</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">n_components</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">)</span>
</pre></div>
</div>
<p>if pre-computed distances are to be used,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sce</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">palantir</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">n_components</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">knn</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">use_adjacency_matrix</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">distances_key</span><span class="o">=</span><span class="s2">&quot;distances&quot;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Visualizing Palantir results</strong></p>
<p><em>tSNE visualization</em></p>
<p>important for Palantir!</p>
<p>Palantir constructs the tSNE map in the embedded space since these maps better
represent the differentiation trajectories.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">tsne</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">n_pcs</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">use_rep</span><span class="o">=</span><span class="s1">&#39;X_palantir_multiscale&#39;</span><span class="p">,</span> <span class="n">perplexity</span><span class="o">=</span><span class="mi">150</span><span class="p">)</span>
</pre></div>
</div>
<p><em>tsne by cell size</em></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">tsne</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;n_counts&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Imputed gene expression visualized on tSNE maps</em></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">tsne</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gene_symbols</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;CD34&#39;</span><span class="p">,</span> <span class="s1">&#39;MPO&#39;</span><span class="p">,</span> <span class="s1">&#39;GATA1&#39;</span><span class="p">,</span> <span class="s1">&#39;IRF8&#39;</span><span class="p">],</span>
<span class="gp">... </span>    <span class="n">layer</span><span class="o">=</span><span class="s1">&#39;palantir_imp&#39;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">color</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;CD34&#39;</span><span class="p">,</span> <span class="s1">&#39;MPO&#39;</span><span class="p">,</span> <span class="s1">&#39;GATA1&#39;</span><span class="p">,</span> <span class="s1">&#39;IRF8&#39;</span><span class="p">]</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Running Palantir</strong></p>
<p>Palantir can be run by specifying an approximate early cell. While Palantir
automatically determines the terminal states, they can also be specified using the
<code class="docutils literal notranslate"><span class="pre">termine_states</span></code> parameter.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">start_cell</span> <span class="o">=</span> <span class="s1">&#39;Run5_164698952452459&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pr_res</span> <span class="o">=</span> <span class="n">sce</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">palantir_results</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">early_cell</span><span class="o">=</span><span class="n">start_cell</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ms_data</span><span class="o">=</span><span class="s1">&#39;X_palantir_multiscale&#39;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">num_waypoints</span><span class="o">=</span><span class="mi">500</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>A <code class="docutils literal notranslate"><span class="pre">start_cell</span></code> must be defined for every data set. The start cell for
this dataset was chosen based on high expression of CD34.</p>
</div>
<p>At this point the returned Palantir object <code class="docutils literal notranslate"><span class="pre">pr_res</span></code> can be used for all downstream
analysis and plotting. Please consult this notebook
<a class="reference external" href="https://github.com/dpeerlab/Palantir/blob/master/notebooks/Palantir_sample_notebook.ipynb">Palantir_sample_notebook.ipynb</a>.
It provides a comprehensive guide to draw <em>gene expression trends</em>, amongst other
things.</p>
</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="scanpy.external.tl.phate.html" class="btn btn-neutral float-left" title="scanpy.external.tl.phate" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="scanpy.external.tl.trimap.html" class="btn btn-neutral float-right" title="scanpy.external.tl.trimap" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Alex Wolf, Philipp Angerer, Fidel Ramirez, Isaac Virshup, Sergei Rybakov, Gokcen Eraslan, Tom White, Malte Luecken, Davide Cittaro, Tobias Callies, Marius Lange, Andr√©s R. Mu√±oz-Rojas..
      <span class="commit">Revision <code>8736fc3b</code>.
      </span></p>
  </div>

   

</footer>
        </div>
      </div>
    </section>
  </div>
  

  <div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="Versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      <span class="fa fa-book"> Read the Docs</span>
      v: stable
      <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
      <dl>
        <dt>Versions</dt>
        
          <dd><a href="/en/latest/">latest</a></dd>
        
          <dd><a href="/en/stable/">stable</a></dd>
        
          <dd><a href="/en/docsearch/">docsearch</a></dd>
        
      </dl>
      <dl>
        <dt>Downloads</dt>
        
      </dl>
      <dl>
        
        <dt>On Read the Docs</dt>
          <dd>
            <a href="//readthedocs.com/projects/icb-scanpy/?fromdocs=icb-scanpy">Project Home</a>
          </dd>
          <dd>
            <a href="//readthedocs.com/builds/icb-scanpy/?fromdocs=icb-scanpy">Builds</a>
          </dd>
      </dl>
    </div>
  </div>
<script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>