<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />

  <title>Release notes &mdash; Scanpy 1.9.0.dev72+g7b22b756 documentation</title>

<link rel=stylesheet href="https://cdn.jsdelivr.net/npm/docsearch.js@2.6/dist/cdn/docsearch.min.css"/>

      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/scanpy.css" type="text/css" />
      <link rel="stylesheet" href="../_static/plot_directive.css" type="text/css" />
      <link rel="stylesheet" href="../_static/typehints.css" type="text/css" />
      <link rel="stylesheet" href="https://assets.readthedocs.com/css/badge_only.css" type="text/css" />
    <link rel="canonical" href="https://scanpy.readthedocs.io/en/stable/release-notes/index.html" />

  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script async="async" src="https://assets.readthedocs.com/javascript/readthedocs-doc-embed.js"></script>
    <script src="../_static/js/theme.js"></script>

<script src="https://cdn.jsdelivr.net/npm/docsearch.js@2.6/dist/cdn/docsearch.min.js"></script>
<script>
  document.addEventListener('DOMContentLoaded', () => docsearch({
    apiKey: 'fa4304eb95d2134997e3729553a674b2',
    indexName: 'scanpy',
    inputSelector: '#rtd-search-form > input[name="q"]',
    algoliaOptions: {
      facetFilters: ['version:latest'],
      hitsPerPage: 10,
    },
    debug: false,
  }))
</script>


    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Community" href="../community.html" />
    <link rel="prev" title="Ecosystem" href="../ecosystem.html" />
 



<!-- RTD Extra Head -->

<link rel="stylesheet" href="https://assets.readthedocs.com/css/readthedocs-doc-embed.css" type="text/css" />

<script type="application/json" id="READTHEDOCS_DATA">{"ad_free": false, "api_host": "https://readthedocs.com", "build_date": "2022-03-03T15:00:21Z", "builder": "sphinx", "canonical_url": null, "commit": "7b22b756", "docroot": "/docs/", "features": {"docsearch_disabled": false}, "global_analytics_code": "UA-17997319-2", "language": "en", "page": "release-notes/index", "programming_language": "py", "project": "icb-scanpy", "proxied_api_host": "/_", "source_suffix": ".rst", "subprojects": {}, "theme": "scanpydoc", "user_analytics_code": "", "version": "latest"}</script>

<!--
Using this variable directly instead of using `JSON.parse` is deprecated.
The READTHEDOCS_DATA global variable will be removed in the future.
-->
<script type="text/javascript">
READTHEDOCS_DATA = JSON.parse(document.getElementById('READTHEDOCS_DATA').innerHTML);
</script>

<script type="text/javascript" src="https://assets.readthedocs.com/javascript/readthedocs-analytics.js" async="async"></script>

<!-- end RTD <extrahead> -->
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html">
            <img src="../_static/Scanpy_Logo_BrightFG.svg" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                latest
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../usage-principles.html">Usage Principles</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api.html">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../external.html">External API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ecosystem.html">Ecosystem</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Release notes</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#version-1-9">Version 1.9</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#the-future">1.9.0 <small>the future</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-8">Version 1.8</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id1">1.8.3 <small>the future</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id2">1.8.2 <small>2021-11-3</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3">1.8.1 <small>2021-07-07</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id4">1.8.0 <small>2021-06-28</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-7">Version 1.7</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id5">1.7.2 <small>2021-04-07</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id6">1.7.1 <small>2021-02-24</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id7">1.7.0 <small>2021-02-03</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-6">Version 1.6</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id11">1.6.0 <small>2020-08-15</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-5">Version 1.5</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id12">1.5.1 <small>2020-05-21</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id13">1.5.0 <small>2020-05-15</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-4">Version 1.4</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id14">1.4.6 <small>2020-03-17</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id16">1.4.5 <small>2019-12-30</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id17">1.4.4 <small>2019-07-20</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id18">1.4.3 <small>2019-05-14</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id19">1.4.2 <small>2019-05-06</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id20">1.4.1 <small>2019-04-26</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-3">Version 1.3</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id22">1.3.6 <small>2018-12-11</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id23">1.3.5 <small>2018-12-09</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id24">1.3.4 <small>2018-11-24</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id28">1.3.3 <small>2018-11-05</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id29">1.3.1 <small>2018-09-03</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-2">Version 1.2</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id36">1.2.1 <small>2018-06-08</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id38">1.2.0 <small>2018-06-08</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-1">Version 1.1</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id39">1.1.0 <small>2018-06-01</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-1-0">Version 1.0</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id47">1.0.0 <small>2018-03-30</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-0-4">Version 0.4</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id51">0.4.3 <small>2018-02-09</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id53">0.4.2 <small>2018-01-07</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id55">0.4.0 <small>2017-12-23</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-0-3">Version 0.3</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id57">0.3.2 <small>2017-11-29</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id58">0.3.0 <small>2017-11-16</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-0-2">Version 0.2</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id60">0.2.9 <small>2017-10-25</small></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id62">0.2.1 <small>2017-07-24</small></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#version-0-1">Version 0.1</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id63">0.1.0 <small>2017-05-17</small></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../community.html">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../news.html">News</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dev/index.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Scanpy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Release notes</li>
      <li class="wy-breadcrumbs-aside">
              <a href="https://github.com/theislab/scanpy/blob/master/docs/release-notes/index.rst" class="fa fa-github"> Edit on GitHub</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="release-notes">
<h1>Release notes<a class="headerlink" href="#release-notes" title="Permalink to this headline"></a></h1>
<section id="version-1-9">
<h2>Version 1.9<a class="headerlink" href="#version-1-9" title="Permalink to this headline"></a></h2>
<section id="the-future">
<h3>1.9.0 <small>the future</small><a class="headerlink" href="#the-future" title="Permalink to this headline"></a></h3>
<p class="rubric">Features</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.filter_rank_genes_groups.html#scanpy.tl.filter_rank_genes_groups" title="scanpy.tl.filter_rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">filter_rank_genes_groups()</span></code></a> now allows to filter with absolute values of log fold change <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1649">PR 1649</a> <span class="smaller">S Rybakov</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.embedding_density.html#scanpy.pl.embedding_density" title="scanpy.pl.embedding_density"><code class="xref py py-func docutils literal notranslate"><span class="pre">embedding_density()</span></code></a> now allows more than 10 groups <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1936">PR 1936</a> <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
</section>
<section id="version-1-8">
<h2>Version 1.8<a class="headerlink" href="#version-1-8" title="Permalink to this headline"></a></h2>
<section id="id1">
<h3>1.8.3 <small>the future</small><a class="headerlink" href="#id1" title="Permalink to this headline"></a></h3>
<p class="rubric">Docs</p>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Fixed finding variables with <code class="docutils literal notranslate"><span class="pre">use_raw=True</span></code> and <code class="docutils literal notranslate"><span class="pre">basis=None</span></code> in <a class="reference internal" href="../generated/scanpy.pl.scatter.html#scanpy.pl.scatter" title="scanpy.pl.scatter"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pl.scatter()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/2027">PR 2027</a> <small>E Rice</small></p></li>
<li><p>Fixed <a class="reference internal" href="../generated/scanpy.external.pp.scrublet.html#scanpy.external.pp.scrublet" title="scanpy.external.pp.scrublet"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.external.pp.scrublet()</span></code></a> to address <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/1957">issue 1957</a> <span class="smaller">FlMai</span> and ensure raw counts are used for simulation</p></li>
<li><p>Functions in <a class="reference internal" href="../api.html#module-scanpy.datasets" title="scanpy.datasets"><code class="xref py py-mod docutils literal notranslate"><span class="pre">scanpy.datasets</span></code></a> no longer throw <code class="docutils literal notranslate"><span class="pre">OldFormatWarnings</span></code> when using <code class="docutils literal notranslate"><span class="pre">anndata</span></code> <code class="docutils literal notranslate"><span class="pre">0.8</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/2096">PR 2096</a> <small>I Virshup</small></p></li>
<li><p>Fixed use of <a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pp.neighbors()</span></code></a> with <code class="docutils literal notranslate"><span class="pre">method='rapids'</span></code>: RAPIDS cuML no longer returns a squared Euclidean distance matrix, so we should not square-root the kNN distance matrix. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1828">PR 1828</a> <small>M Zaslavsky</small></p></li>
<li><p>Removed <code class="docutils literal notranslate"><span class="pre">pytables</span></code> dependency by implementing <code class="docutils literal notranslate"><span class="pre">read_10x_h5</span></code> with <code class="docutils literal notranslate"><span class="pre">h5py</span></code> due to installation errors on Windows <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/2064">PR 2064</a></p></li>
</ul>
<p class="rubric">Performance</p>
<p class="rubric">Ecosystem</p>
</section>
<section id="id2">
<h3>1.8.2 <small>2021-11-3</small><a class="headerlink" href="#id2" title="Permalink to this headline"></a></h3>
<p class="rubric">Docs</p>
<ul class="simple">
<li><p>Update conda installation instructions <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1974">PR 1974</a> <span class="smaller">L Heumos</span></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Fix plotting after <a class="reference internal" href="../generated/scanpy.tl.filter_rank_genes_groups.html#scanpy.tl.filter_rank_genes_groups" title="scanpy.tl.filter_rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.filter_rank_genes_groups()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1942">PR 1942</a> <span class="smaller">S Rybakov</span></p></li>
<li><p>Fix <code class="docutils literal notranslate"><span class="pre">use_raw=None</span></code> using <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.var_names.html#anndata.AnnData.var_names" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">anndata.AnnData.var_names</span></code></a> if <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.raw.html#anndata.AnnData.raw" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">anndata.AnnData.raw</span></code></a>
is present in <a class="reference internal" href="../generated/scanpy.tl.score_genes.html#scanpy.tl.score_genes" title="scanpy.tl.score_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.score_genes()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1999">PR 1999</a> <span class="smaller">M Klein</span></p></li>
<li><p>Fix compatibility with UMAP 0.5.2 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/2028">PR 2028</a> <span class="smaller">L Mcinnes</span></p></li>
<li><p>Fixed non-determinism in <a class="reference internal" href="../generated/scanpy.pl.paga.html#scanpy.pl.paga" title="scanpy.pl.paga"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pl.paga()</span></code></a> node positions <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1922">PR 1922</a> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">Ecosystem</p>
<ul class="simple">
<li><p>Added PASTE (a tool to align and integrate spatial transcriptomics data) to scanpy ecosystem.</p></li>
</ul>
</section>
<section id="id3">
<h3>1.8.1 <small>2021-07-07</small><a class="headerlink" href="#id3" title="Permalink to this headline"></a></h3>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Fixed reproducibility of <a class="reference internal" href="../generated/scanpy.tl.score_genes.html#scanpy.tl.score_genes" title="scanpy.tl.score_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.score_genes()</span></code></a>. Calculation and output is now float64 type.  <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1890">PR 1890</a> <span class="smaller">I Kucinski</span></p></li>
<li><p>Workarounds for some changes/ bugs in pandas 1.3 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1918">PR 1918</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed bug where <code class="docutils literal notranslate"><span class="pre">sc.pl.paga_compare</span></code> could mislabel nodes on the paga graph <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1898">PR 1898</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed handling of <code class="docutils literal notranslate"><span class="pre">use_raw</span></code> with <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.rank_genes_groups()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1934">PR 1934</a> <span class="smaller">I Virshup</span></p></li>
</ul>
</section>
<section id="id4">
<h3>1.8.0 <small>2021-06-28</small><a class="headerlink" href="#id4" title="Permalink to this headline"></a></h3>
<p class="rubric">Metrics module</p>
<ul>
<li><p>Added <a class="reference internal" href="../api.html#module-scanpy.metrics" title="scanpy.metrics"><code class="xref py py-mod docutils literal notranslate"><span class="pre">scanpy.metrics</span></code></a> module!</p>
<blockquote>
<div><ul class="simple">
<li><p>Added <a class="reference internal" href="../generated/scanpy.metrics.gearys_c.html#scanpy.metrics.gearys_c" title="scanpy.metrics.gearys_c"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.metrics.gearys_c()</span></code></a> for spatial autocorrelation <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/915">PR 915</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Added <a class="reference internal" href="../generated/scanpy.metrics.morans_i.html#scanpy.metrics.morans_i" title="scanpy.metrics.morans_i"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.metrics.morans_i()</span></code></a> for global spatial autocorrelation <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1740">PR 1740</a> <span class="smaller">I Virshup, G Palla</span></p></li>
<li><p>Added <a class="reference internal" href="../generated/scanpy.metrics.confusion_matrix.html#scanpy.metrics.confusion_matrix" title="scanpy.metrics.confusion_matrix"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.metrics.confusion_matrix()</span></code></a> for comparing labellings <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/915">PR 915</a> <span class="smaller">I Virshup</span></p></li>
</ul>
</div></blockquote>
</li>
</ul>
<p class="rubric">Features</p>
<ul class="simple">
<li><p>Added <code class="docutils literal notranslate"><span class="pre">layer</span></code> and <code class="docutils literal notranslate"><span class="pre">copy</span></code> kwargs to <a class="reference internal" href="../generated/scanpy.pp.normalize_total.html#scanpy.pp.normalize_total" title="scanpy.pp.normalize_total"><code class="xref py py-func docutils literal notranslate"><span class="pre">normalize_total()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1667">PR 1667</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">vcenter</span></code> and <code class="docutils literal notranslate"><span class="pre">norm</span></code> arguments to the plotting functions <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1551">PR 1551</a> <span class="smaller">G Eraslan</span></p></li>
<li><p>Standardized and expanded available arguments to the <code class="docutils literal notranslate"><span class="pre">sc.pl.rank_genes_groups*</span></code> family of functions. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1529">PR 1529</a> <span class="smaller">F Ramirez</span> <span class="smaller">I Virshup</span>
- See examples sections of <a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_dotplot.html#scanpy.pl.rank_genes_groups_dotplot" title="scanpy.pl.rank_genes_groups_dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_dotplot()</span></code></a> and <a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_matrixplot.html#scanpy.pl.rank_genes_groups_matrixplot" title="scanpy.pl.rank_genes_groups_matrixplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_matrixplot()</span></code></a> for demonstrations.</p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.tsne.html#scanpy.tl.tsne" title="scanpy.tl.tsne"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.tsne()</span></code></a> now supports the metric argument and records the passed parameters <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1854">PR 1854</a> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">Ecosystem</p>
<ul class="simple">
<li><p>Added <a class="reference external" href="https://github.com/connerlambden/Cube">Cubé</a> to ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1878">PR 1878</a> <span class="smaller">C Lambden</span></p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">triku</span></code> a feature selection method to the ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1722">PR 1722</a> <span class="smaller">AM Ascensión</span></p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">dorothea</span></code> and <code class="docutils literal notranslate"><span class="pre">progeny</span></code> to the ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1767">PR 1767</a> <span class="smaller">P Badia-i-Mompel</span></p></li>
</ul>
<p class="rubric">Documentation</p>
<ul class="simple">
<li><p>Added <a class="reference internal" href="../community.html"><span class="doc">Community</span></a> page to docs <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1856">PR 1856</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Added rendered examples to many plotting functions <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/1664">issue 1664</a> <span class="smaller">A Schaar</span> <span class="smaller">L Zappia</span> <span class="smaller">bio-la</span> <span class="smaller">L Hetzel</span> <span class="smaller">L Dony</span> <span class="smaller">M Buttner</span> <span class="smaller">K Hrovatin</span> <span class="smaller">F Ramirez</span> <span class="smaller">I Virshup</span> <span class="smaller">LouisK92</span> <span class="smaller">mayarali</span></p></li>
<li><p>Integrated <a class="reference external" href="https://docsearch.algolia.com/">DocSearch</a>, a find-as-you-type documentation index search. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1754">PR 1754</a> <span class="smaller">P Angerer</span></p></li>
<li><ul>
<li><p>Reorganized reference docs <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1753">PR 1753</a> <span class="smaller">I Virshup</span></p></li>
</ul>
</li>
<li><p>Clarified docs issues for <a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a>,
<a class="reference internal" href="../generated/scanpy.tl.diffmap.html#scanpy.tl.diffmap" title="scanpy.tl.diffmap"><code class="xref py py-func docutils literal notranslate"><span class="pre">diffmap()</span></code></a>, <a class="reference internal" href="../generated/scanpy.pp.calculate_qc_metrics.html#scanpy.pp.calculate_qc_metrics" title="scanpy.pp.calculate_qc_metrics"><code class="xref py py-func docutils literal notranslate"><span class="pre">calculate_qc_metrics()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1680">PR 1680</a> <span class="smaller">G Palla</span></p></li>
<li><p>Fixed typos in grouped plot doc-strings <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1877">PR 1877</a> <span class="smaller">C Rands</span></p></li>
<li><p>Extended examples for differential expression plotting. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1529">PR 1529</a> <span class="smaller">F Ramirez</span>
- See <a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_dotplot.html#scanpy.pl.rank_genes_groups_dotplot" title="scanpy.pl.rank_genes_groups_dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_dotplot()</span></code></a> or <a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_matrixplot.html#scanpy.pl.rank_genes_groups_matrixplot" title="scanpy.pl.rank_genes_groups_matrixplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_matrixplot()</span></code></a> for examples.</p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Fix <a class="reference internal" href="../generated/scanpy.pl.paga_path.html#scanpy.pl.paga_path" title="scanpy.pl.paga_path"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pl.paga_path()</span></code></a> <code class="docutils literal notranslate"><span class="pre">TypeError</span></code> with recent versions of anndata <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1047">PR 1047</a> <span class="smaller">P Angerer</span></p></li>
<li><p>Fix detection of whether IPython is running <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1844">PR 1844</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed reproducibility of <a class="reference internal" href="../generated/scanpy.tl.diffmap.html#scanpy.tl.diffmap" title="scanpy.tl.diffmap"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.diffmap()</span></code></a> (added random_state) <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1858">PR 1858</a> <span class="smaller">I Kucinski</span></p></li>
<li><p>Fixed errors and warnings from embedding plots with small numbers of categories after <code class="docutils literal notranslate"><span class="pre">sns.set_palette</span></code> was called <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1886">PR 1886</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed handling of <code class="docutils literal notranslate"><span class="pre">gene_symbols</span></code> argument in a number of <code class="docutils literal notranslate"><span class="pre">sc.pl.rank_genes_groups*</span></code> functions <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1529">PR 1529</a> <span class="smaller">F Ramirez</span> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed handling of <code class="docutils literal notranslate"><span class="pre">use_raw</span></code> for <code class="docutils literal notranslate"><span class="pre">sc.tl.rank_genes_groups</span></code> when no <code class="docutils literal notranslate"><span class="pre">.raw</span></code> is present <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1895">PR 1895</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_violin.html#scanpy.pl.rank_genes_groups_violin" title="scanpy.pl.rank_genes_groups_violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pl.rank_genes_groups_violin()</span></code></a> now works for <code class="docutils literal notranslate"><span class="pre">raw=False</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1669">PR 1669</a> <span class="smaller">M van den Beek</span></p></li>
</ul>
<p class="rubric">Development processes</p>
<ul class="simple">
<li><p>Switched to <a class="reference external" href="https://flit.readthedocs.io/en/latest/">flit</a> for building and deploying the package, a simple tool with an easy to understand command line interface and metadata <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1527">PR 1527</a> <span class="smaller">P Angerer</span></p></li>
<li><p>Use <a class="reference external" href="https://pre-commit.com">pre-commit</a> for style checks <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1684">PR 1684</a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1848">PR 1848</a> <span class="smaller">L Heumos</span> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">Deprecations</p>
<ul class="simple">
<li><p>Dropped support for Python 3.6. <a class="reference external" href="https://numpy.org/neps/nep-0029-deprecation_policy.html">More details here</a>. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1897">PR 1897</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Deprecated <code class="docutils literal notranslate"><span class="pre">layers</span></code> and <code class="docutils literal notranslate"><span class="pre">layers_norm</span></code> kwargs to <a class="reference internal" href="../generated/scanpy.pp.normalize_total.html#scanpy.pp.normalize_total" title="scanpy.pp.normalize_total"><code class="xref py py-func docutils literal notranslate"><span class="pre">normalize_total()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1667">PR 1667</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Deprecated <code class="docutils literal notranslate"><span class="pre">MulticoreTSNE</span></code> backend for <a class="reference internal" href="../generated/scanpy.tl.tsne.html#scanpy.tl.tsne" title="scanpy.tl.tsne"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.tl.tsne()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1854">PR 1854</a> <span class="smaller">I Virshup</span></p></li>
</ul>
</section>
</section>
<section id="version-1-7">
<h2>Version 1.7<a class="headerlink" href="#version-1-7" title="Permalink to this headline"></a></h2>
<section id="id5">
<h3>1.7.2 <small>2021-04-07</small><a class="headerlink" href="#id5" title="Permalink to this headline"></a></h3>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.logging.print_versions.html#scanpy.logging.print_versions" title="scanpy.logging.print_versions"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.logging.print_versions()</span></code></a> now works when <code class="docutils literal notranslate"><span class="pre">python&lt;3.8</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1691">PR 1691</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.regress_out.html#scanpy.pp.regress_out" title="scanpy.pp.regress_out"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pp.regress_out()</span></code></a> now uses <code class="docutils literal notranslate"><span class="pre">joblib</span></code> as the parallel backend, and should stop oversubscribing threads <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1694">PR 1694</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes" title="scanpy.pp.highly_variable_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pp.highly_variable_genes()</span></code></a> with <code class="docutils literal notranslate"><span class="pre">flavor=&quot;seurat_v3&quot;</span></code> now returns correct gene means and -variances when used with <code class="docutils literal notranslate"><span class="pre">batch_key</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1732">PR 1732</a> <span class="smaller">J Lause</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes" title="scanpy.pp.highly_variable_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pp.highly_variable_genes()</span></code></a> now throws a warning instead of an error when non-integer values are passed for method <code class="docutils literal notranslate"><span class="pre">&quot;seurat_v3&quot;</span></code>. The check can be skipped by passing <code class="docutils literal notranslate"><span class="pre">check_values=False</span></code>. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1679">PR 1679</a> <span class="smaller">G Palla</span></p></li>
</ul>
<p class="rubric">Ecosystem</p>
<ul class="simple">
<li><p>Added <code class="docutils literal notranslate"><span class="pre">triku</span></code> a feature selection method to the ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1722">PR 1722</a> <span class="smaller">AM Ascensión</span></p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">dorothea</span></code> and <code class="docutils literal notranslate"><span class="pre">progeny</span></code> to the ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1767">PR 1767</a> <span class="smaller">P Badia-i-Mompel</span></p></li>
</ul>
</section>
<section id="id6">
<h3>1.7.1 <small>2021-02-24</small><a class="headerlink" href="#id6" title="Permalink to this headline"></a></h3>
<p class="rubric">Documentation</p>
<ul class="simple">
<li><p>More twitter handles for core devs <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1676">PR 1676</a> <span class="smaller">G Eraslan</span></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.dendrogram.html#scanpy.tl.dendrogram" title="scanpy.tl.dendrogram"><code class="xref py py-func docutils literal notranslate"><span class="pre">dendrogram()</span></code></a> use <code class="docutils literal notranslate"><span class="pre">1</span> <span class="pre">-</span> <span class="pre">correlation</span></code> as distance matrix to compute the dendrogram <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1614">PR 1614</a> <span class="smaller">F Ramirez</span></p></li>
<li><p>Fixed <a class="reference internal" href="../generated/scanpy.get.obs_df.html#scanpy.get.obs_df" title="scanpy.get.obs_df"><code class="xref py py-func docutils literal notranslate"><span class="pre">obs_df()</span></code></a>/ <a class="reference internal" href="../generated/scanpy.get.var_df.html#scanpy.get.var_df" title="scanpy.get.var_df"><code class="xref py py-func docutils literal notranslate"><span class="pre">var_df()</span></code></a> erroring when <code class="docutils literal notranslate"><span class="pre">keys</span></code> not passed <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1637">PR 1637</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed argument handling for <a class="reference internal" href="../generated/scanpy.external.pp.scrublet.html#scanpy.external.pp.scrublet" title="scanpy.external.pp.scrublet"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.external.pp.scrublet()</span></code></a> <span class="smaller">J Manning</span></p></li>
<li><p>Fixed passing of <code class="docutils literal notranslate"><span class="pre">kwargs</span></code> to <a class="reference internal" href="../generated/scanpy.pl.violin.html#scanpy.pl.violin" title="scanpy.pl.violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pl.violin()</span></code></a> when <code class="docutils literal notranslate"><span class="pre">stripplot</span></code> was also used <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1655">PR 1655</a> <span class="smaller">M van den Beek</span></p></li>
<li><p>Fixed colorbar creation in <code class="docutils literal notranslate"><span class="pre">scanpy.pl.timeseries_as_heatmap</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1654">PR 1654</a> <span class="smaller">M van den Beek</span></p></li>
</ul>
</section>
<section id="id7">
<h3>1.7.0 <small>2021-02-03</small><a class="headerlink" href="#id7" title="Permalink to this headline"></a></h3>
<p class="rubric">Features</p>
<ul class="simple">
<li><p>Add new 10x Visium datasets to <a class="reference internal" href="../generated/scanpy.datasets.visium_sge.html#scanpy.datasets.visium_sge" title="scanpy.datasets.visium_sge"><code class="xref py py-func docutils literal notranslate"><span class="pre">visium_sge()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1473">PR 1473</a> <span class="smaller">G Palla</span></p></li>
<li><p>Enable download of source image for 10x visium datasets in <a class="reference internal" href="../generated/scanpy.datasets.visium_sge.html#scanpy.datasets.visium_sge" title="scanpy.datasets.visium_sge"><code class="xref py py-func docutils literal notranslate"><span class="pre">visium_sge()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1506">PR 1506</a> <span class="smaller">H Spitzer</span></p></li>
<li><p>Refactor of <a class="reference internal" href="../generated/scanpy.pl.spatial.html#scanpy.pl.spatial" title="scanpy.pl.spatial"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.pl.spatial()</span></code></a>. Better support for plotting without an image, as well as directly providing images <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1512">PR 1512</a> <span class="smaller">G Palla</span></p></li>
<li><p>Dict input for <a class="reference internal" href="../generated/scanpy.queries.enrich.html#scanpy.queries.enrich" title="scanpy.queries.enrich"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.queries.enrich()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1488">PR 1488</a> <span class="smaller">G Eraslan</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.get.rank_genes_groups_df.html#scanpy.get.rank_genes_groups_df" title="scanpy.get.rank_genes_groups_df"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_df()</span></code></a> can now return fraction of cells in a group expressing a gene, and allows retrieving values for multiple groups at once <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1388">PR 1388</a> <span class="smaller">G Eraslan</span></p></li>
<li><p>Color annotations for gene sets in <a class="reference internal" href="../generated/scanpy.pl.heatmap.html#scanpy.pl.heatmap" title="scanpy.pl.heatmap"><code class="xref py py-func docutils literal notranslate"><span class="pre">heatmap()</span></code></a> are now matched to color for cluster <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1511">PR 1511</a> <span class="smaller">L Sikkema</span></p></li>
<li><p>PCA plots can now annotate axes with variance explained <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1470">PR 1470</a> <span class="smaller">bfurtwa</span></p></li>
<li><p>Plots with <code class="docutils literal notranslate"><span class="pre">groupby</span></code> arguments can now group by values in the index by passing the index’s name (like <code class="docutils literal notranslate"><span class="pre">pd.DataFrame.groupby</span></code>). <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1583">PR 1583</a> <span class="smaller">F Ramirez</span></p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">na_color</span></code> and <code class="docutils literal notranslate"><span class="pre">na_in_legend</span></code> keyword arguments to <a class="reference internal" href="../generated/scanpy.pl.embedding.html#scanpy.pl.embedding" title="scanpy.pl.embedding"><code class="xref py py-func docutils literal notranslate"><span class="pre">embedding()</span></code></a> plots. Allows specifying color for missing or filtered values in plots like <a class="reference internal" href="../generated/scanpy.pl.umap.html#scanpy.pl.umap" title="scanpy.pl.umap"><code class="xref py py-func docutils literal notranslate"><span class="pre">umap()</span></code></a> or <a class="reference internal" href="../generated/scanpy.pl.spatial.html#scanpy.pl.spatial" title="scanpy.pl.spatial"><code class="xref py py-func docutils literal notranslate"><span class="pre">spatial()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1356">PR 1356</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.embedding.html#scanpy.pl.embedding" title="scanpy.pl.embedding"><code class="xref py py-func docutils literal notranslate"><span class="pre">embedding()</span></code></a> plots now support passing <code class="docutils literal notranslate"><span class="pre">dict</span></code> of <code class="docutils literal notranslate"><span class="pre">{cluster_name:</span> <span class="pre">cluster_color,</span> <span class="pre">...}</span></code> for palette argument  <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1392">PR 1392</a> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">External tools (new)</p>
<ul class="simple">
<li><p>Add <a class="reference external" href="https://github.com/brianhie/scanorama">Scanorama</a> integration to scanpy external API (<a class="reference internal" href="../generated/scanpy.external.pp.scanorama_integrate.html#scanpy.external.pp.scanorama_integrate" title="scanpy.external.pp.scanorama_integrate"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanorama_integrate()</span></code></a>) <a class="reference internal" href="../references.html#hie19" id="id8"><span>[Hie19]</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1332">PR 1332</a> <span class="smaller">B Hie</span></p></li>
<li><p>Scrublet <a class="reference internal" href="../references.html#wolock19" id="id9"><span>[Wolock19]</span></a> integration: <a class="reference internal" href="../generated/scanpy.external.pp.scrublet.html#scanpy.external.pp.scrublet" title="scanpy.external.pp.scrublet"><code class="xref py py-func docutils literal notranslate"><span class="pre">scrublet()</span></code></a>, <a class="reference internal" href="../generated/scanpy.external.pp.scrublet_simulate_doublets.html#scanpy.external.pp.scrublet_simulate_doublets" title="scanpy.external.pp.scrublet_simulate_doublets"><code class="xref py py-func docutils literal notranslate"><span class="pre">scrublet_simulate_doublets()</span></code></a>, and plotting method <a class="reference internal" href="../generated/scanpy.external.pl.scrublet_score_distribution.html#scanpy.external.pl.scrublet_score_distribution" title="scanpy.external.pl.scrublet_score_distribution"><code class="xref py py-func docutils literal notranslate"><span class="pre">scrublet_score_distribution()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1476">PR 1476</a> <span class="smaller">J Manning</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.pp.hashsolo.html#scanpy.external.pp.hashsolo" title="scanpy.external.pp.hashsolo"><code class="xref py py-func docutils literal notranslate"><span class="pre">hashsolo()</span></code></a> for HTO demultiplexing <a class="reference internal" href="../references.html#bernstein20" id="id10"><span>[Bernstein20]</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1432">PR 1432</a> <span class="smaller">NJ Bernstein</span></p></li>
<li><p>Added <a class="reference external" href="https://github.com/icbi-lab/scirpy">scirpy</a> (sc-AIRR analysis) to ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1453">PR 1453</a> <span class="smaller">G Sturm</span></p></li>
<li><p>Added <a class="reference external" href="https://scvi-tools.org">scvi-tools</a> to ecosystem page <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1421">PR 1421</a> <span class="smaller">A Gayoso</span></p></li>
</ul>
<p class="rubric">External tools (changes)</p>
<ul class="simple">
<li><p>Updates for <a class="reference internal" href="../generated/scanpy.external.tl.palantir.html#scanpy.external.tl.palantir" title="scanpy.external.tl.palantir"><code class="xref py py-func docutils literal notranslate"><span class="pre">palantir()</span></code></a> and <a class="reference internal" href="../generated/scanpy.external.tl.palantir_results.html#scanpy.external.tl.palantir_results" title="scanpy.external.tl.palantir_results"><code class="xref py py-func docutils literal notranslate"><span class="pre">palantir_results()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1245">PR 1245</a> <span class="smaller">A Mousa</span></p></li>
<li><p>Fixes to <a class="reference internal" href="../generated/scanpy.external.tl.harmony_timeseries.html#scanpy.external.tl.harmony_timeseries" title="scanpy.external.tl.harmony_timeseries"><code class="xref py py-func docutils literal notranslate"><span class="pre">harmony_timeseries()</span></code></a> docs <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1248">PR 1248</a> <span class="smaller">A Mousa</span></p></li>
<li><p>Support for <code class="docutils literal notranslate"><span class="pre">leiden</span></code> clustering by <a class="reference internal" href="../generated/scanpy.external.tl.phenograph.html#scanpy.external.tl.phenograph" title="scanpy.external.tl.phenograph"><code class="xref py py-func docutils literal notranslate"><span class="pre">scanpy.external.tl.phenograph()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1080">PR 1080</a> <span class="smaller">A Mousa</span></p></li>
<li><p>Deprecate <code class="docutils literal notranslate"><span class="pre">scanpy.external.pp.scvi</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1554">PR 1554</a> <span class="smaller">G Xing</span></p></li>
<li><p>Updated default params of <a class="reference internal" href="../generated/scanpy.external.tl.sam.html#scanpy.external.tl.sam" title="scanpy.external.tl.sam"><code class="xref py py-func docutils literal notranslate"><span class="pre">sam()</span></code></a> to work with larger data <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1540">PR 1540</a> <span class="smaller">A Tarashansky</span></p></li>
</ul>
<p class="rubric">Documentation</p>
<ul class="simple">
<li><p><a class="reference internal" href="../dev/index.html#contribution-guide"><span class="std std-ref">New contribution guide</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1544">PR 1544</a> <span class="smaller">I Virshup</span></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">zsh</span></code> installation instructions <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1444">PR 1444</a> <span class="smaller">P Angerer</span></p></li>
</ul>
<p class="rubric">Performance</p>
<ul class="simple">
<li><p>Speed up <a class="reference internal" href="../generated/scanpy.read_10x_h5.html#scanpy.read_10x_h5" title="scanpy.read_10x_h5"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_10x_h5()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1402">PR 1402</a> <span class="smaller">P Weiler</span></p></li>
<li><p>Speed ups for <a class="reference internal" href="../generated/scanpy.get.obs_df.html#scanpy.get.obs_df" title="scanpy.get.obs_df"><code class="xref py py-func docutils literal notranslate"><span class="pre">obs_df()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1499">PR 1499</a> <span class="smaller">F Ramirez</span></p></li>
</ul>
<p class="rubric">Bugfixes</p>
<ul class="simple">
<li><p>Consistent fold-change, fractions calculation for filter_rank_genes_groups <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1391">PR 1391</a> <span class="smaller">S Rybakov</span></p></li>
<li><p>Fixed bug where <code class="docutils literal notranslate"><span class="pre">score_genes</span></code> would error if one gene was passed <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1398">PR 1398</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed <code class="docutils literal notranslate"><span class="pre">log1p</span></code> inplace on integer dense arrays <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1400">PR 1400</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fix docstring formatting for <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1417">PR 1417</a> <span class="smaller">P Weiler</span></p></li>
<li><p>Removed <code class="docutils literal notranslate"><span class="pre">PendingDeprecationWarning`s</span> <span class="pre">from</span> <span class="pre">use</span> <span class="pre">of</span> <span class="pre">`np.matrix</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1424">PR 1424</a> <span class="smaller">P Weiler</span></p></li>
<li><p>Fixed indexing byg in <code class="docutils literal notranslate"><span class="pre">~scanpy.pp.highly_variable_genes</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1456">PR 1456</a> <span class="smaller">V Bergen</span></p></li>
<li><p>Fix default number of genes for marker_genes_overlap <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1464">PR 1464</a> <span class="smaller">MD Luecken</span></p></li>
<li><p>Fixed passing <code class="docutils literal notranslate"><span class="pre">groupby</span></code> and <code class="docutils literal notranslate"><span class="pre">dendrogram_key</span></code> to <a class="reference internal" href="../generated/scanpy.tl.dendrogram.html#scanpy.tl.dendrogram" title="scanpy.tl.dendrogram"><code class="xref py py-func docutils literal notranslate"><span class="pre">dendrogram()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1465">PR 1465</a> <span class="smaller">M Varma</span></p></li>
<li><p>Fixed download path of <code class="docutils literal notranslate"><span class="pre">pbmc3k_processed</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1472">PR 1472</a> <span class="smaller">D Strobl</span></p></li>
<li><p>Better error message when computing DE with a group of size 1 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1490">PR 1490</a> <span class="smaller">J Manning</span></p></li>
<li><p>Update cugraph API usage for v0.16 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1494">PR 1494</a> <span class="smaller">R Ilango</span></p></li>
<li><p>Fixed <code class="docutils literal notranslate"><span class="pre">marker_gene_overlap</span></code> default value for <code class="docutils literal notranslate"><span class="pre">top_n_markers</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1464">PR 1464</a> <span class="smaller">MD Luecken</span></p></li>
<li><p>Pass <code class="docutils literal notranslate"><span class="pre">random_state</span></code> to RAPIDs UMAP <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1474">PR 1474</a> <span class="smaller">C Nolet</span></p></li>
<li><p>Fixed <code class="docutils literal notranslate"><span class="pre">anndata</span></code> version requirement for <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.concat.html#anndata.concat" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-func docutils literal notranslate"><span class="pre">concat()</span></code></a> (re-exported from scanpy as <code class="docutils literal notranslate"><span class="pre">sc.concat</span></code>) <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1491">PR 1491</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed the width of the progress bar when downloading data <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1507">PR 1507</a> <span class="smaller">M Klein</span></p></li>
<li><p>Updated link for <code class="docutils literal notranslate"><span class="pre">moignard15</span></code> dataset <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1542">PR 1542</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed bug where calling <code class="docutils literal notranslate"><span class="pre">set_figure_params</span></code> could block if IPython was installed, but not used. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1547">PR 1547</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.violin.html#scanpy.pl.violin" title="scanpy.pl.violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">violin()</span></code></a> no longer fails if <code class="docutils literal notranslate"><span class="pre">.raw</span></code> not present <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1548">PR 1548</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.spatial.html#scanpy.pl.spatial" title="scanpy.pl.spatial"><code class="xref py py-func docutils literal notranslate"><span class="pre">spatial()</span></code></a> refactoring and better handling of spatial data <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1512">PR 1512</a> <span class="smaller">G Palla</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.pca.html#scanpy.pp.pca" title="scanpy.pp.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a> works with <code class="docutils literal notranslate"><span class="pre">chunked=True</span></code> again <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1592">PR 1592</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.ingest.html#scanpy.tl.ingest" title="scanpy.tl.ingest"><code class="xref py py-func docutils literal notranslate"><span class="pre">ingest()</span></code></a> now works with umap-learn 0.5.0 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1601">PR 1601</a> <span class="smaller">S Rybakov</span></p></li>
</ul>
</section>
</section>
<section id="version-1-6">
<h2>Version 1.6<a class="headerlink" href="#version-1-6" title="Permalink to this headline"></a></h2>
<section id="id11">
<h3>1.6.0 <small>2020-08-15</small><a class="headerlink" href="#id11" title="Permalink to this headline"></a></h3>
<p>This release includes an overhaul of <a class="reference internal" href="../generated/scanpy.pl.dotplot.html#scanpy.pl.dotplot" title="scanpy.pl.dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">dotplot()</span></code></a>, <a class="reference internal" href="../generated/scanpy.pl.matrixplot.html#scanpy.pl.matrixplot" title="scanpy.pl.matrixplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">matrixplot()</span></code></a>, and <a class="reference internal" href="../generated/scanpy.pl.stacked_violin.html#scanpy.pl.stacked_violin" title="scanpy.pl.stacked_violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">stacked_violin()</span></code></a> (<a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1210">PR 1210</a> <span class="smaller">F Ramirez</span>), and of the internals of <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> (<a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1156">PR 1156</a> <span class="smaller">S Rybakov</span>).</p>
<p class="rubric">Overhaul of <a class="reference internal" href="../generated/scanpy.pl.dotplot.html#scanpy.pl.dotplot" title="scanpy.pl.dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">dotplot()</span></code></a>, <a class="reference internal" href="../generated/scanpy.pl.matrixplot.html#scanpy.pl.matrixplot" title="scanpy.pl.matrixplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">matrixplot()</span></code></a>, and <a class="reference internal" href="../generated/scanpy.pl.stacked_violin.html#scanpy.pl.stacked_violin" title="scanpy.pl.stacked_violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">stacked_violin()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1210">PR 1210</a> <span class="smaller">F Ramirez</span></p>
<ul>
<li><p>An overhauled tutorial <a class="tutorial reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/plotting/core.html">→ tutorial: plotting/core</a>.</p></li>
<li><p>New plotting classes can be accessed directly (e.g., <a class="reference internal" href="../generated/classes/scanpy.pl.DotPlot.html#scanpy.pl.DotPlot" title="scanpy.pl.DotPlot"><code class="xref py py-class docutils literal notranslate"><span class="pre">DotPlot</span></code></a>) or using the <code class="docutils literal notranslate"><span class="pre">return_fig</span></code> param.</p></li>
<li><p>It is possible to plot log fold change and p-values in the <a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_dotplot.html#scanpy.pl.rank_genes_groups_dotplot" title="scanpy.pl.rank_genes_groups_dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_dotplot()</span></code></a> family of functions.</p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">ax</span></code> parameter which allows embedding the plot in other images.</p></li>
<li><p>Added option to include a bar plot instead of the dendrogram containing the cell/observation totals per category.</p></li>
<li><p>Return a dictionary of axes for further manipulation. This includes the main plot, legend and dendrogram to totals</p></li>
<li><p>Legends can be removed.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">groupby</span></code> param can take a list of categories, e.g., <code class="docutils literal notranslate"><span class="pre">groupby=[‘tissue’,</span> <span class="pre">‘cell</span> <span class="pre">type’]</span></code>.</p></li>
<li><p>Added padding parameter to <code class="docutils literal notranslate"><span class="pre">dotplot</span></code> and <code class="docutils literal notranslate"><span class="pre">stacked_violin</span></code>. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1270">PR 1270</a></p></li>
<li><p>Added title for colorbar and positioned as in dotplot for <a class="reference internal" href="../generated/scanpy.pl.matrixplot.html#scanpy.pl.matrixplot" title="scanpy.pl.matrixplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">matrixplot()</span></code></a>.</p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.dotplot.html#scanpy.pl.dotplot" title="scanpy.pl.dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">dotplot()</span></code></a> changes:</p>
<blockquote>
<div><ul class="simple">
<li><p>Improved the colorbar and size legend for dotplots. Now the colorbar and size have titles, which can be modified using the <code class="docutils literal notranslate"><span class="pre">colorbar_title</span></code> and <code class="docutils literal notranslate"><span class="pre">size_title</span></code> params. They also align at the bottom of the image and do not shrink if the dotplot image is smaller.</p></li>
<li><p>Allow plotting genes in rows and categories in columns (<code class="docutils literal notranslate"><span class="pre">swap_axes</span></code>).</p></li>
<li><p>Using <a class="reference internal" href="../generated/classes/scanpy.pl.DotPlot.html#scanpy.pl.DotPlot" title="scanpy.pl.DotPlot"><code class="xref py py-class docutils literal notranslate"><span class="pre">DotPlot</span></code></a>, the <code class="docutils literal notranslate"><span class="pre">dot_edge_color</span></code> and line width can be modified, a grid can be added, and other modifications are enabled.</p></li>
<li><p>A new style was added in which the dots are replaced by an empty circle and the square behind the circle is colored (like in matrixplots).</p></li>
</ul>
</div></blockquote>
</li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.stacked_violin.html#scanpy.pl.stacked_violin" title="scanpy.pl.stacked_violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">stacked_violin()</span></code></a> changes:</p>
<blockquote>
<div><ul class="simple">
<li><p>Violin colors can be colored based on average gene expression as in dotplots.</p></li>
<li><p>The linewidth of the violin plots is thinner.</p></li>
<li><p>Removed the tics for the y-axis as they tend to overlap with each other. Using the style method they can be displayed if needed.</p></li>
</ul>
</div></blockquote>
</li>
</ul>
<p class="rubric">Additions</p>
<ul class="simple">
<li><p><a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.concat.html#anndata.concat" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-func docutils literal notranslate"><span class="pre">concat()</span></code></a> is now exported from scanpy, see <a class="reference external" href="https://anndata.readthedocs.io/en/stable/concatenation.html" title="(in anndata v0.7.7.dev1+g33006dc)"><span>Concatenation</span></a> for more info. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1338">PR 1338</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Added highly variable gene selection strategy from Seurat v3 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1204">PR 1204</a> <span class="smaller">A Gayoso</span></p></li>
<li><p>Added <a class="reference external" href="https://github.com/theislab/cellrank/">CellRank</a> to scanpy ecosystem <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1304">PR 1304</a> <span class="smaller">giovp</span></p></li>
<li><p>Added <code class="docutils literal notranslate"><span class="pre">backup_url</span></code> param to <a class="reference internal" href="../generated/scanpy.read_10x_h5.html#scanpy.read_10x_h5" title="scanpy.read_10x_h5"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_10x_h5()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1296">PR 1296</a> <span class="smaller">A Gayoso</span></p></li>
<li><p>Allow prefix for <a class="reference internal" href="../generated/scanpy.read_10x_mtx.html#scanpy.read_10x_mtx" title="scanpy.read_10x_mtx"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_10x_mtx()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1250">PR 1250</a>  <span class="smaller">G Sturm</span></p></li>
<li><p>Optional tie correction for the <code class="docutils literal notranslate"><span class="pre">'wilcoxon'</span></code> method in <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1330">PR 1330</a>  <span class="smaller">S Rybakov</span></p></li>
<li><p>Use <code class="docutils literal notranslate"><span class="pre">sinfo</span></code> for <a class="reference internal" href="../generated/scanpy.logging.print_versions.html#scanpy.logging.print_versions" title="scanpy.logging.print_versions"><code class="xref py py-func docutils literal notranslate"><span class="pre">print_versions()</span></code></a> and add <a class="reference internal" href="../generated/scanpy.logging.print_header.html#scanpy.logging.print_header" title="scanpy.logging.print_header"><code class="xref py py-func docutils literal notranslate"><span class="pre">print_header()</span></code></a> to do what it previously did. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1338">PR 1338</a> <span class="smaller">I Virshup</span> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1373">PR 1373</a></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Avoid warning in <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> if ‘t-test’ is passed <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1303">PR 1303</a> <span class="smaller">A Wolf</span></p></li>
<li><p>Restrict sphinx version to &lt;3.1, &gt;3.0 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1297">PR 1297</a>  <span class="smaller">I Virshup</span></p></li>
<li><p>Clean up <code class="docutils literal notranslate"><span class="pre">_ranks</span></code> and fix <code class="docutils literal notranslate"><span class="pre">dendrogram</span></code> for scipy 1.5 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1290">PR 1290</a> <span class="smaller">S Rybakov</span></p></li>
<li><p>Use <code class="docutils literal notranslate"><span class="pre">.raw</span></code> to translate gene symbols if applicable <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1278">PR 1278</a> <span class="smaller">E Rice</span></p></li>
<li><p>Fix <code class="docutils literal notranslate"><span class="pre">diffmap</span></code> (<a class="issue reference external" href="https://github.com/theislab/scanpy/issues/1262">issue 1262</a>) <span class="smaller">G Eraslan</span></p></li>
<li><p>Fix <code class="docutils literal notranslate"><span class="pre">neighbors</span></code> in <code class="docutils literal notranslate"><span class="pre">spring_project</span></code> <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/1260">issue 1260</a>  <span class="smaller">S Rybakov</span></p></li>
<li><p>Fix default size of dot in spatial plots <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1255">PR 1255</a> <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/1253">issue 1253</a> <span class="smaller">giovp</span></p></li>
<li><p>Bumped version requirement of <code class="docutils literal notranslate"><span class="pre">scipy</span></code> to <code class="docutils literal notranslate"><span class="pre">scipy&gt;1.4</span></code> to support <code class="docutils literal notranslate"><span class="pre">rmatmat</span></code> argument of <code class="docutils literal notranslate"><span class="pre">LinearOperator</span></code> <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/1246">issue 1246</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fix asymmetry of scores for the <code class="docutils literal notranslate"><span class="pre">'wilcoxon'</span></code> method in <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/754">issue 754</a>  <span class="smaller">S Rybakov</span></p></li>
<li><p>Avoid trimming of gene names in <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/753">issue 753</a>  <span class="smaller">S Rybakov</span></p></li>
</ul>
</section>
</section>
<section id="version-1-5">
<h2>Version 1.5<a class="headerlink" href="#version-1-5" title="Permalink to this headline"></a></h2>
<section id="id12">
<h3>1.5.1 <small>2020-05-21</small><a class="headerlink" href="#id12" title="Permalink to this headline"></a></h3>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Fixed a bug in <a class="reference internal" href="../generated/scanpy.pp.pca.html#scanpy.pp.pca" title="scanpy.pp.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a>, where <code class="docutils literal notranslate"><span class="pre">random_state</span></code> did not have an effect for sparse input <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1240">PR 1240</a> <span class="smaller">I Virshup</span></p></li>
<li><p>Fixed docstring in <a class="reference internal" href="../generated/scanpy.pp.pca.html#scanpy.pp.pca" title="scanpy.pp.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a> which included an unused argument <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1240">PR 1240</a> <span class="smaller">I Virshup</span></p></li>
</ul>
</section>
<section id="id13">
<h3>1.5.0 <small>2020-05-15</small><a class="headerlink" href="#id13" title="Permalink to this headline"></a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">1.5.0</span></code> release adds a lot of new functionality, much of which takes advantage of <a class="reference external" href="https://anndata.readthedocs.io/en/stable/api.html#module-anndata" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-mod docutils literal notranslate"><span class="pre">anndata</span></code></a> updates <code class="docutils literal notranslate"><span class="pre">0.7.0</span> <span class="pre">-</span> <span class="pre">0.7.2</span></code>. Highlights of this release include support for spatial data, dedicated handling of graphs in AnnData, sparse PCA, an interface with scvi, and others.</p>
<p class="rubric">Spatial data support</p>
<ul class="simple">
<li><p>Basic analysis <a class="tutorial reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/spatial/basic-analysis.html">→ tutorial: spatial/basic-analysis</a> and integration with single cell data <a class="tutorial reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/spatial/integration-scanorama.html">→ tutorial: spatial/integration-scanorama</a> <span class="smaller">G Palla</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.read_visium.html#scanpy.read_visium" title="scanpy.read_visium"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_visium()</span></code></a> read 10x Visium data <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1034">PR 1034</a> <span class="smaller">G Palla, P Angerer, I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.datasets.visium_sge.html#scanpy.datasets.visium_sge" title="scanpy.datasets.visium_sge"><code class="xref py py-func docutils literal notranslate"><span class="pre">visium_sge()</span></code></a> load Visium data directly from 10x Genomics <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1013">PR 1013</a> <span class="smaller">M Mirkazemi, G Palla, P Angerer</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.spatial.html#scanpy.pl.spatial" title="scanpy.pl.spatial"><code class="xref py py-func docutils literal notranslate"><span class="pre">spatial()</span></code></a> plot spatial data <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1012">PR 1012</a> <span class="smaller">G Palla, P Angerer</span></p></li>
</ul>
<p class="rubric">New functionality</p>
<ul class="simple">
<li><p>Many functions, like <a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a> and <a class="reference internal" href="../generated/scanpy.tl.umap.html#scanpy.tl.umap" title="scanpy.tl.umap"><code class="xref py py-func docutils literal notranslate"><span class="pre">umap()</span></code></a>, now store cell-by-cell graphs in <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.obsp.html#anndata.AnnData.obsp" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">obsp</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1118">PR 1118</a> <span class="smaller">S Rybakov</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.scale.html#scanpy.pp.scale" title="scanpy.pp.scale"><code class="xref py py-func docutils literal notranslate"><span class="pre">scale()</span></code></a> and <a class="reference internal" href="../generated/scanpy.pp.log1p.html#scanpy.pp.log1p" title="scanpy.pp.log1p"><code class="xref py py-func docutils literal notranslate"><span class="pre">log1p()</span></code></a> can be used on any element in <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.layers.html#anndata.AnnData.layers" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">layers</span></code></a> or <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.obsm.html#anndata.AnnData.obsm" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">obsm</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1173">PR 1173</a> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">External tools</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">scanpy.external.pp.scvi</span></code> for preprocessing with scVI <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1085">PR 1085</a> <span class="smaller">G Xing</span></p></li>
<li><p>Guide for using <a class="reference internal" href="../tutorials.html#conversion-to-r"><span class="std std-ref">Scanpy in R</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1186">PR 1186</a> <span class="smaller">L Zappia</span></p></li>
</ul>
<p class="rubric">Performance</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.pca.html#scanpy.pp.pca" title="scanpy.pp.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a> now uses efficient implicit centering for sparse matrices. This can lead to signifigantly improved performance for large datasets <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1066">PR 1066</a> <span class="smaller">A Tarashansky</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.score_genes.html#scanpy.tl.score_genes" title="scanpy.tl.score_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">score_genes()</span></code></a> now has an efficient implementation for sparse matrices with missing values <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1196">PR 1196</a> <span class="smaller">redst4r</span>.</p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The new <a class="reference internal" href="../generated/scanpy.pp.pca.html#scanpy.pp.pca" title="scanpy.pp.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a> implementation can result in slightly different results for sparse matrices. See the pr (<a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1066">PR 1066</a>) and documentation for more info.</p>
</div>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pl.stacked_violin.html#scanpy.pl.stacked_violin" title="scanpy.pl.stacked_violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">stacked_violin()</span></code></a> can now be used as a subplot <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1084">PR 1084</a> <span class="smaller">P Angerer</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.score_genes.html#scanpy.tl.score_genes" title="scanpy.tl.score_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">score_genes()</span></code></a> has improved logging <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1119">PR 1119</a> <span class="smaller">G Eraslan</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.scale.html#scanpy.pp.scale" title="scanpy.pp.scale"><code class="xref py py-func docutils literal notranslate"><span class="pre">scale()</span></code></a> now saves mean and standard deviation in the <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.var.html#anndata.AnnData.var" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-attr docutils literal notranslate"><span class="pre">var</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1173">PR 1173</a> <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.harmony_timeseries.html#scanpy.external.tl.harmony_timeseries" title="scanpy.external.tl.harmony_timeseries"><code class="xref py py-func docutils literal notranslate"><span class="pre">harmony_timeseries()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1091">PR 1091</a> <span class="smaller">A Mousa</span></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.combat.html#scanpy.pp.combat" title="scanpy.pp.combat"><code class="xref py py-func docutils literal notranslate"><span class="pre">combat()</span></code></a> now works when <code class="docutils literal notranslate"><span class="pre">obs_names</span></code> aren’t unique. <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1215">PR 1215</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.scale.html#scanpy.pp.scale" title="scanpy.pp.scale"><code class="xref py py-func docutils literal notranslate"><span class="pre">scale()</span></code></a> can now be used on dense arrays without centering <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1160">PR 1160</a> <span class="smaller">simonwm</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.regress_out.html#scanpy.pp.regress_out" title="scanpy.pp.regress_out"><code class="xref py py-func docutils literal notranslate"><span class="pre">regress_out()</span></code></a> now works when some features are constant <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1194">PR 1194</a> <span class="smaller">simonwm</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.normalize_total.html#scanpy.pp.normalize_total" title="scanpy.pp.normalize_total"><code class="xref py py-func docutils literal notranslate"><span class="pre">normalize_total()</span></code></a> errored if the passed object was a view <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1200">PR 1200</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a> sometimes ignored the <code class="docutils literal notranslate"><span class="pre">n_pcs</span></code> param <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1124">PR 1124</a> <span class="smaller">V Bergen</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.datasets.ebi_expression_atlas.html#scanpy.datasets.ebi_expression_atlas" title="scanpy.datasets.ebi_expression_atlas"><code class="xref py py-func docutils literal notranslate"><span class="pre">ebi_expression_atlas()</span></code></a> which contained some out-of-date URLs <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1102">PR 1102</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.ingest.html#scanpy.tl.ingest" title="scanpy.tl.ingest"><code class="xref py py-func docutils literal notranslate"><span class="pre">ingest()</span></code></a> for UMAP <code class="docutils literal notranslate"><span class="pre">0.4</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1165">PR 1165</a> <span class="smaller">S Rybakov</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.louvain.html#scanpy.tl.louvain" title="scanpy.tl.louvain"><code class="xref py py-func docutils literal notranslate"><span class="pre">louvain()</span></code></a> for Louvain <code class="docutils literal notranslate"><span class="pre">0.6</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1197">PR 1197</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes" title="scanpy.pp.highly_variable_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">highly_variable_genes()</span></code></a> which could lead to incorrect results when the <code class="docutils literal notranslate"><span class="pre">batch_key</span></code> argument was used <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1180">PR 1180</a> <span class="smaller">G Eraslan</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.ingest.html#scanpy.tl.ingest" title="scanpy.tl.ingest"><code class="xref py py-func docutils literal notranslate"><span class="pre">ingest()</span></code></a> where an inconsistent number of neighbors was used <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1111">PR 1111</a> <span class="smaller">S Rybakov</span></p></li>
</ul>
</section>
</section>
<section id="version-1-4">
<h2>Version 1.4<a class="headerlink" href="#version-1-4" title="Permalink to this headline"></a></h2>
<section id="id14">
<h3>1.4.6 <small>2020-03-17</small><a class="headerlink" href="#id14" title="Permalink to this headline"></a></h3>
<p class="rubric">Functionality in <code class="docutils literal notranslate"><span class="pre">external</span></code></p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.sam.html#scanpy.external.tl.sam" title="scanpy.external.tl.sam"><code class="xref py py-func docutils literal notranslate"><span class="pre">sam()</span></code></a> self-assembling manifolds <a class="reference internal" href="../references.html#tarashansky19" id="id15"><span>[Tarashansky19]</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/903">PR 903</a> <span class="smaller">A Tarashansky</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.harmony_timeseries.html#scanpy.external.tl.harmony_timeseries" title="scanpy.external.tl.harmony_timeseries"><code class="xref py py-func docutils literal notranslate"><span class="pre">harmony_timeseries()</span></code></a> for trajectory inference on discrete time points <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/994">PR 994</a> <span class="smaller">A Mousa</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.wishbone.html#scanpy.external.tl.wishbone" title="scanpy.external.tl.wishbone"><code class="xref py py-func docutils literal notranslate"><span class="pre">wishbone()</span></code></a> for trajectory inference (bifurcations) <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1063">PR 1063</a> <span class="smaller">A Mousa</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pl.violin.html#scanpy.pl.violin" title="scanpy.pl.violin"><code class="xref py py-mod docutils literal notranslate"><span class="pre">violin</span></code></a> now reads <code class="docutils literal notranslate"><span class="pre">.uns['colors_...']</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1029">PR 1029</a> <span class="smaller">michalk8</span></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>adapt <a class="reference internal" href="../generated/scanpy.tl.ingest.html#scanpy.tl.ingest" title="scanpy.tl.ingest"><code class="xref py py-func docutils literal notranslate"><span class="pre">ingest()</span></code></a> for UMAP 0.4 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1038">PR 1038</a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1106">PR 1106</a> <span class="smaller">S Rybakov</span></p></li>
<li><p>compat with matplotlib 3.1 and 3.2 <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1090">PR 1090</a> <span class="smaller">I Virshup, P Angerer</span></p></li>
<li><p>fix PAGA for new igraph <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1037">PR 1037</a> <span class="smaller">P Angerer</span></p></li>
<li><p>fix rapids compat of louvain <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/1079">PR 1079</a> <span class="smaller">LouisFaure</span></p></li>
</ul>
</section>
<section id="id16">
<h3>1.4.5 <small>2019-12-30</small><a class="headerlink" href="#id16" title="Permalink to this headline"></a></h3>
<p>Please install <code class="docutils literal notranslate"><span class="pre">scanpy==1.4.5.post3</span></code> instead of <code class="docutils literal notranslate"><span class="pre">scanpy==1.4.5</span></code>.</p>
<p class="rubric">New functionality</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.ingest.html#scanpy.tl.ingest" title="scanpy.tl.ingest"><code class="xref py py-func docutils literal notranslate"><span class="pre">ingest()</span></code></a> maps labels and embeddings of reference data to new data <a class="tutorial reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/integrating-data-using-ingest.html">→ tutorial: integrating-data-using-ingest</a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/651">PR 651</a> <span class="smaller">S Rybakov, A Wolf</span></p></li>
<li><p><a class="reference internal" href="../api.html#module-scanpy.queries" title="scanpy.queries"><code class="xref py py-mod docutils literal notranslate"><span class="pre">queries</span></code></a> recieved many updates including enrichment through <a class="reference external" href="https://biit.cs.ut.ee/gprofiler/">gprofiler</a> and more advanced biomart queries <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/467">PR 467</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.set_figure_params.html#scanpy.set_figure_params" title="scanpy.set_figure_params"><code class="xref py py-func docutils literal notranslate"><span class="pre">set_figure_params()</span></code></a> allows setting <code class="docutils literal notranslate"><span class="pre">figsize</span></code> and accepts <code class="docutils literal notranslate"><span class="pre">facecolor='white'</span></code>, useful for working in dark mode  <span class="smaller">A Wolf</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.downsample_counts.html#scanpy.pp.downsample_counts" title="scanpy.pp.downsample_counts"><code class="xref py py-mod docutils literal notranslate"><span class="pre">downsample_counts</span></code></a> now always preserves the dtype of it’s input, instead of converting floats to ints <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/865">PR 865</a> <span class="smaller">I Virshup</span></p></li>
<li><p>allow specifying a base for <a class="reference internal" href="../generated/scanpy.pp.log1p.html#scanpy.pp.log1p" title="scanpy.pp.log1p"><code class="xref py py-func docutils literal notranslate"><span class="pre">log1p()</span></code></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/931">PR 931</a> <span class="smaller">G Eraslan</span></p></li>
<li><p>run neighbors on a GPU using rapids <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/830">PR 830</a> <span class="smaller">T White</span></p></li>
<li><p>param docs from typed params <span class="smaller">P Angerer</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.embedding_density.html#scanpy.tl.embedding_density" title="scanpy.tl.embedding_density"><code class="xref py py-func docutils literal notranslate"><span class="pre">embedding_density()</span></code></a> now only takes one positional argument; similar for <a class="reference internal" href="../generated/scanpy.pl.embedding_density.html#scanpy.pl.embedding_density" title="scanpy.pl.embedding_density"><code class="xref py py-func docutils literal notranslate"><span class="pre">embedding_density()</span></code></a>, which gains a param <code class="docutils literal notranslate"><span class="pre">groupby</span></code> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/965">PR 965</a> <span class="smaller">A Wolf</span></p></li>
<li><p>webpage overhaul, ecosystem page, release notes, tutorials overhaul <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/960">PR 960</a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/966">PR 966</a> <span class="smaller">A Wolf</span></p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<ul class="simple">
<li><p>changed default <code class="docutils literal notranslate"><span class="pre">solver</span></code> in <a class="reference internal" href="../generated/scanpy.tl.pca.html#scanpy.tl.pca" title="scanpy.tl.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a> from <code class="docutils literal notranslate"><span class="pre">auto</span></code> to <code class="docutils literal notranslate"><span class="pre">arpack</span></code></p></li>
<li><p>changed default <code class="docutils literal notranslate"><span class="pre">use_raw</span></code> in <a class="reference internal" href="../generated/scanpy.tl.score_genes.html#scanpy.tl.score_genes" title="scanpy.tl.score_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">score_genes()</span></code></a> from <code class="docutils literal notranslate"><span class="pre">False</span></code> to <code class="docutils literal notranslate"><span class="pre">None</span></code></p></li>
</ul>
</div>
</section>
<section id="id17">
<h3>1.4.4 <small>2019-07-20</small><a class="headerlink" href="#id17" title="Permalink to this headline"></a></h3>
<p class="rubric">New functionality</p>
<ul class="simple">
<li><p><a class="reference internal" href="../api.html#module-scanpy.get" title="scanpy.get"><code class="xref py py-mod docutils literal notranslate"><span class="pre">scanpy.get</span></code></a> adds helper functions for extracting data in convenient formats <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/619">PR 619</a> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p>Stopped deprecations warnings from AnnData <code class="docutils literal notranslate"><span class="pre">0.6.22</span></code> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.normalize_total.html#scanpy.pp.normalize_total" title="scanpy.pp.normalize_total"><code class="xref py py-func docutils literal notranslate"><span class="pre">normalize_total()</span></code></a> gains param <code class="docutils literal notranslate"><span class="pre">exclude_highly_expressed</span></code>, and <code class="docutils literal notranslate"><span class="pre">fraction</span></code> is renamed to <code class="docutils literal notranslate"><span class="pre">max_fraction</span></code> with better docs <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
<section id="id18">
<h3>1.4.3 <small>2019-05-14</small><a class="headerlink" href="#id18" title="Permalink to this headline"></a></h3>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a> correctly infers <code class="docutils literal notranslate"><span class="pre">n_neighbors</span></code> again from <code class="docutils literal notranslate"><span class="pre">params</span></code>, which was temporarily broken in <code class="docutils literal notranslate"><span class="pre">v1.4.2</span></code> <span class="smaller">I Virshup</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.calculate_qc_metrics.html#scanpy.pp.calculate_qc_metrics" title="scanpy.pp.calculate_qc_metrics"><code class="xref py py-func docutils literal notranslate"><span class="pre">calculate_qc_metrics()</span></code></a> is single threaded by default for datasets under 300,000 cells – allowing cached compilation <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/615">PR 615</a> <span class="smaller">I Virshup</span></p></li>
</ul>
</section>
<section id="id19">
<h3>1.4.2 <small>2019-05-06</small><a class="headerlink" href="#id19" title="Permalink to this headline"></a></h3>
<p class="rubric">New functionality</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.combat.html#scanpy.pp.combat" title="scanpy.pp.combat"><code class="xref py py-func docutils literal notranslate"><span class="pre">combat()</span></code></a> supports additional covariates which may include adjustment variables or biological condition <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/618">PR 618</a> <span class="smaller">G Eraslan</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes" title="scanpy.pp.highly_variable_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">highly_variable_genes()</span></code></a> has a <code class="docutils literal notranslate"><span class="pre">batch_key</span></code> option which performs HVG selection in each batch separately to avoid selecting genes that vary strongly across batches <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/622">PR 622</a> <span class="smaller">G Eraslan</span></p></li>
</ul>
<p class="rubric">Bug fixes</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> t-test implementation doesn’t return NaN when variance is 0, also changed to scipy’s implementation <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/621">PR 621</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.umap.html#scanpy.tl.umap" title="scanpy.tl.umap"><code class="xref py py-func docutils literal notranslate"><span class="pre">umap()</span></code></a> with <code class="docutils literal notranslate"><span class="pre">init_pos='paga'</span></code> detects correct <code class="docutils literal notranslate"><span class="pre">dtype</span></code> <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.louvain.html#scanpy.tl.louvain" title="scanpy.tl.louvain"><code class="xref py py-func docutils literal notranslate"><span class="pre">louvain()</span></code></a> and <a class="reference internal" href="../generated/scanpy.tl.leiden.html#scanpy.tl.leiden" title="scanpy.tl.leiden"><code class="xref py py-func docutils literal notranslate"><span class="pre">leiden()</span></code></a> auto-generate <code class="docutils literal notranslate"><span class="pre">key_added=louvain_R</span></code> upon passing <code class="docutils literal notranslate"><span class="pre">restrict_to</span></code>, which was temporarily changed in <code class="docutils literal notranslate"><span class="pre">1.4.1</span></code> <span class="smaller">A Wolf</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a> and <a class="reference internal" href="../generated/scanpy.tl.umap.html#scanpy.tl.umap" title="scanpy.tl.umap"><code class="xref py py-func docutils literal notranslate"><span class="pre">umap()</span></code></a> got rid of UMAP legacy code and introduced UMAP as a dependency <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/576">PR 576</a> <span class="smaller">S Rybakov</span></p></li>
</ul>
</section>
<section id="id20">
<h3>1.4.1 <small>2019-04-26</small><a class="headerlink" href="#id20" title="Permalink to this headline"></a></h3>
<p class="rubric">New functionality</p>
<ul class="simple">
<li><p>Scanpy has a command line interface again. Invoking it with <code class="docutils literal notranslate"><span class="pre">scanpy</span> <span class="pre">somecommand</span> <span class="pre">[args]</span></code> calls <code class="docutils literal notranslate"><span class="pre">scanpy-somecommand</span> <span class="pre">[args]</span></code>, except for builtin commands (currently <code class="docutils literal notranslate"><span class="pre">scanpy</span> <span class="pre">settings</span></code>)  <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/604">PR 604</a> <span class="smaller">P Angerer</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.datasets.ebi_expression_atlas.html#scanpy.datasets.ebi_expression_atlas" title="scanpy.datasets.ebi_expression_atlas"><code class="xref py py-func docutils literal notranslate"><span class="pre">ebi_expression_atlas()</span></code></a> allows convenient download of EBI expression atlas <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.marker_gene_overlap.html#scanpy.tl.marker_gene_overlap" title="scanpy.tl.marker_gene_overlap"><code class="xref py py-func docutils literal notranslate"><span class="pre">marker_gene_overlap()</span></code></a> computes overlaps of marker genes <span class="smaller">M Luecken</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.filter_rank_genes_groups.html#scanpy.tl.filter_rank_genes_groups" title="scanpy.tl.filter_rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">filter_rank_genes_groups()</span></code></a> filters out genes based on fold change and fraction of cells expressing genes <span class="smaller">F Ramirez</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.normalize_total.html#scanpy.pp.normalize_total" title="scanpy.pp.normalize_total"><code class="xref py py-func docutils literal notranslate"><span class="pre">normalize_total()</span></code></a> replaces <a class="reference internal" href="../generated/scanpy.pp.normalize_per_cell.html#scanpy.pp.normalize_per_cell" title="scanpy.pp.normalize_per_cell"><code class="xref py py-func docutils literal notranslate"><span class="pre">normalize_per_cell()</span></code></a>, is more efficient and provides a parameter to only normalize using a fraction of expressed genes <span class="smaller">S Rybakov</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.downsample_counts.html#scanpy.pp.downsample_counts" title="scanpy.pp.downsample_counts"><code class="xref py py-func docutils literal notranslate"><span class="pre">downsample_counts()</span></code></a> has been sped up, changed default value of <code class="docutils literal notranslate"><span class="pre">replace</span></code> parameter to <code class="docutils literal notranslate"><span class="pre">False</span></code>  <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/474">PR 474</a> <span class="smaller">I Virshup</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.embedding_density.html#scanpy.tl.embedding_density" title="scanpy.tl.embedding_density"><code class="xref py py-func docutils literal notranslate"><span class="pre">embedding_density()</span></code></a> computes densities on embeddings  <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/543">PR 543</a> <span class="smaller">M Luecken</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.palantir.html#scanpy.external.tl.palantir" title="scanpy.external.tl.palantir"><code class="xref py py-func docutils literal notranslate"><span class="pre">palantir()</span></code></a> interfaces Palantir <a class="reference internal" href="../references.html#setty18" id="id21"><span>[Setty18]</span></a>  <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/493">PR 493</a> <span class="smaller">A Mousa</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">.layers</span></code> support of scatter plots <span class="smaller">F Ramirez</span></p></li>
<li><p>fix double-logarithmization in compute of log fold change in <a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> <span class="smaller">A Muñoz-Rojas</span></p></li>
<li><p>fix return sections of docs <span class="smaller">P Angerer</span></p></li>
</ul>
</section>
</section>
<section id="version-1-3">
<h2>Version 1.3<a class="headerlink" href="#version-1-3" title="Permalink to this headline"></a></h2>
<section id="id22">
<h3>1.3.6 <small>2018-12-11</small><a class="headerlink" href="#id22" title="Permalink to this headline"></a></h3>
<p class="rubric">Major updates</p>
<ul class="simple">
<li><p>a new plotting gallery for <a class="reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/visualizing-marker-genes.html" title="(in Scanpy)"><span>&lt;no title&gt;</span></a> <span class="smaller">F Ramirez</span></p></li>
<li><p>tutorials are integrated on ReadTheDocs, <a class="reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html" title="(in Scanpy)"><span>Preprocessing and clustering 3k PBMCs</span></a> and <a class="reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/paga-paul15.html" title="(in Scanpy)"><span>Trajectory inference for hematopoiesis in mouse</span></a> <span class="smaller">A Wolf</span></p></li>
</ul>
<p class="rubric">Interactive exploration of analysis results through <em>manifold viewers</em></p>
<ul class="simple">
<li><p>CZI’s <a class="reference external" href="https://github.com/chanzuckerberg/cellxgene">cellxgene</a> directly reads <code class="docutils literal notranslate"><span class="pre">.h5ad</span></code> files <span class="smaller">the cellxgene developers</span></p></li>
<li><p>the <a class="reference external" href="https://github.com/maximilianh/cellBrowser">UCSC Single Cell Browser</a> requires exporting via <a class="reference internal" href="../generated/scanpy.external.exporting.cellbrowser.html#scanpy.external.exporting.cellbrowser" title="scanpy.external.exporting.cellbrowser"><code class="xref py py-func docutils literal notranslate"><span class="pre">cellbrowser()</span></code></a> <span class="smaller">M Haeussler</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes" title="scanpy.pp.highly_variable_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">highly_variable_genes()</span></code></a> supersedes <a class="reference internal" href="../generated/scanpy.pp.filter_genes_dispersion.html#scanpy.pp.filter_genes_dispersion" title="scanpy.pp.filter_genes_dispersion"><code class="xref py py-func docutils literal notranslate"><span class="pre">filter_genes_dispersion()</span></code></a>, it gives the same results but, by default, expects logarithmized data and doesn’t subset <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
<section id="id23">
<h3>1.3.5 <small>2018-12-09</small><a class="headerlink" href="#id23" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>uncountable figure improvements <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/369">PR 369</a> <span class="smaller">F Ramirez</span></p></li>
</ul>
</section>
<section id="id24">
<h3>1.3.4 <small>2018-11-24</small><a class="headerlink" href="#id24" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.leiden.html#scanpy.tl.leiden" title="scanpy.tl.leiden"><code class="xref py py-func docutils literal notranslate"><span class="pre">leiden()</span></code></a> wraps the recent graph clustering package by <a class="reference internal" href="../references.html#traag18" id="id25"><span>[Traag18]</span></a> <span class="smaller">K Polanski</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.pp.bbknn.html#scanpy.external.pp.bbknn" title="scanpy.external.pp.bbknn"><code class="xref py py-func docutils literal notranslate"><span class="pre">bbknn()</span></code></a> wraps the recent batch correction package <a class="reference internal" href="../references.html#polanski19" id="id26"><span>[Polanski19]</span></a> <span class="smaller">K Polanski</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.calculate_qc_metrics.html#scanpy.pp.calculate_qc_metrics" title="scanpy.pp.calculate_qc_metrics"><code class="xref py py-func docutils literal notranslate"><span class="pre">calculate_qc_metrics()</span></code></a> caculates a number of quality control metrics, similar to <code class="docutils literal notranslate"><span class="pre">calculateQCMetrics</span></code> from <em>Scater</em> <a class="reference internal" href="../references.html#mccarthy17" id="id27"><span>[McCarthy17]</span></a> <span class="smaller">I Virshup</span></p></li>
</ul>
</section>
<section id="id28">
<h3>1.3.3 <small>2018-11-05</small><a class="headerlink" href="#id28" title="Permalink to this headline"></a></h3>
<p class="rubric">Major updates</p>
<ul class="simple">
<li><p>a fully distributed preprocessing backend <span class="smaller">T White and the Laserson Lab</span></p></li>
</ul>
<p class="rubric">Code design</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.read_10x_h5.html#scanpy.read_10x_h5" title="scanpy.read_10x_h5"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_10x_h5()</span></code></a> and <a class="reference internal" href="../generated/scanpy.read_10x_mtx.html#scanpy.read_10x_mtx" title="scanpy.read_10x_mtx"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_10x_mtx()</span></code></a> read Cell Ranger 3.0 outputs <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/334">PR 334</a> <span class="smaller">Q Gong</span></p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p class="rubric">Also see changes in anndata 0.6.</p>
<ul class="simple">
<li><p>changed default compression to <code class="docutils literal notranslate"><span class="pre">None</span></code> in <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.write_h5ad.html#anndata.AnnData.write_h5ad" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-meth docutils literal notranslate"><span class="pre">write_h5ad()</span></code></a> to speed up read and write, disk space use is usually less critical</p></li>
<li><p>performance gains in <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.write_h5ad.html#anndata.AnnData.write_h5ad" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-meth docutils literal notranslate"><span class="pre">write_h5ad()</span></code></a> due to better handling of strings and categories <span class="smaller">S Rybakov</span></p></li>
</ul>
</div>
</section>
<section id="id29">
<h3>1.3.1 <small>2018-09-03</small><a class="headerlink" href="#id29" title="Permalink to this headline"></a></h3>
<p class="rubric">RNA velocity in single cells <a class="reference internal" href="../references.html#manno18" id="id30"><span>[Manno18]</span></a></p>
<ul class="simple">
<li><p>Scanpy and AnnData support loom’s layers so that computations for single-cell RNA velocity <a class="reference internal" href="../references.html#manno18" id="id31"><span>[Manno18]</span></a> become feasible <span class="smaller">S Rybakov and V Bergen</span></p></li>
<li><p><a class="reference external" href="https://github.com/theislab/scvelo">scvelo</a> harmonizes with Scanpy and is able to process loom files with splicing information produced by Velocyto <a class="reference internal" href="../references.html#manno18" id="id32"><span>[Manno18]</span></a>, it runs a lot faster than the count matrix analysis of Velocyto and provides several conceptual developments</p></li>
</ul>
<p class="rubric">Plotting (<a class="reference internal" href="../api.html#pl-generic"><span class="std std-ref">Generic</span></a>)</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pl.dotplot.html#scanpy.pl.dotplot" title="scanpy.pl.dotplot"><code class="xref py py-func docutils literal notranslate"><span class="pre">dotplot()</span></code></a> for visualizing genes across conditions and clusters, see <a class="reference external" href="https://gist.github.com/fidelram/2289b7a8d6da055fb058ac9a79ed485c">here</a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/199">PR 199</a> <span class="smaller">F Ramirez</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.heatmap.html#scanpy.pl.heatmap" title="scanpy.pl.heatmap"><code class="xref py py-func docutils literal notranslate"><span class="pre">heatmap()</span></code></a> for pretty heatmaps <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/175">PR 175</a> <span class="smaller">F Ramirez</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.violin.html#scanpy.pl.violin" title="scanpy.pl.violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">violin()</span></code></a> produces very compact overview figures with many panels <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/175">PR 175</a> <span class="smaller">F Ramirez</span></p></li>
</ul>
<p class="rubric">There now is a section on imputation in <a class="reference internal" href="../external.html"><span class="doc">external</span></a>:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.external.pp.magic.html#scanpy.external.pp.magic" title="scanpy.external.pp.magic"><code class="xref py py-func docutils literal notranslate"><span class="pre">magic()</span></code></a> for imputation using data diffusion <a class="reference internal" href="../references.html#vandijk18" id="id34"><span>[vanDijk18]</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/187">PR 187</a> <span class="smaller">S Gigante</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.pp.dca.html#scanpy.external.pp.dca" title="scanpy.external.pp.dca"><code class="xref py py-func docutils literal notranslate"><span class="pre">dca()</span></code></a> for imputation and latent space construction using an autoencoder <a class="reference internal" href="../references.html#eraslan18" id="id35"><span>[Eraslan18]</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/186">PR 186</a> <span class="smaller">G Eraslan</span></p></li>
</ul>
</section>
</section>
<section id="version-1-2">
<h2>Version 1.2<a class="headerlink" href="#version-1-2" title="Permalink to this headline"></a></h2>
<section id="id36">
<h3>1.2.1 <small>2018-06-08</small><a class="headerlink" href="#id36" title="Permalink to this headline"></a></h3>
<p class="rubric">Plotting of <a class="reference internal" href="../api.html#pl-generic"><span class="std std-ref">Generic</span></a> marker genes and quality control.</p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pl.highest_expr_genes.html#scanpy.pl.highest_expr_genes" title="scanpy.pl.highest_expr_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">highest_expr_genes()</span></code></a> for quality control; plot genes with highest mean fraction of cells, similar to <code class="docutils literal notranslate"><span class="pre">plotQC</span></code> of <em>Scater</em> <a class="reference internal" href="../references.html#mccarthy17" id="id37"><span>[McCarthy17]</span></a> <a class="pr reference external" href="https://github.com/theislab/scanpy/pull/169">PR 169</a> <span class="smaller">F Ramirez</span></p></li>
</ul>
</section>
<section id="id38">
<h3>1.2.0 <small>2018-06-08</small><a class="headerlink" href="#id38" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.paga.html#scanpy.tl.paga" title="scanpy.tl.paga"><code class="xref py py-func docutils literal notranslate"><span class="pre">paga()</span></code></a> improved, see <a class="reference external" href="https://github.com/theislab/paga">PAGA</a>; the default model changed, restore the previous default model by passing <code class="docutils literal notranslate"><span class="pre">model='v1.0'</span></code></p></li>
</ul>
</section>
</section>
<section id="version-1-1">
<h2>Version 1.1<a class="headerlink" href="#version-1-1" title="Permalink to this headline"></a></h2>
<section id="id39">
<h3>1.1.0 <small>2018-06-01</small><a class="headerlink" href="#id39" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.set_figure_params.html#scanpy.set_figure_params" title="scanpy.set_figure_params"><code class="xref py py-func docutils literal notranslate"><span class="pre">set_figure_params()</span></code></a> by default passes <code class="docutils literal notranslate"><span class="pre">vector_friendly=True</span></code> and allows you to produce reasonablly sized pdfs by rasterizing large scatter plots <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.draw_graph.html#scanpy.tl.draw_graph" title="scanpy.tl.draw_graph"><code class="xref py py-func docutils literal notranslate"><span class="pre">draw_graph()</span></code></a> defaults to the ForceAtlas2 layout <a class="reference internal" href="../references.html#jacomy14" id="id40"><span>[Jacomy14]</span></a> <a class="reference internal" href="../references.html#chippada18" id="id41"><span>[Chippada18]</span></a>, which is often more visually appealing and whose computation is much faster <span class="smaller">S Wollock</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.scatter.html#scanpy.pl.scatter" title="scanpy.pl.scatter"><code class="xref py py-func docutils literal notranslate"><span class="pre">scatter()</span></code></a> also plots along variables axis <span class="smaller">MD Luecken</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.pca.html#scanpy.pp.pca" title="scanpy.pp.pca"><code class="xref py py-func docutils literal notranslate"><span class="pre">pca()</span></code></a> and <a class="reference internal" href="../generated/scanpy.pp.log1p.html#scanpy.pp.log1p" title="scanpy.pp.log1p"><code class="xref py py-func docutils literal notranslate"><span class="pre">log1p()</span></code></a> support chunk processing <span class="smaller">S Rybakov</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.regress_out.html#scanpy.pp.regress_out" title="scanpy.pp.regress_out"><code class="xref py py-func docutils literal notranslate"><span class="pre">regress_out()</span></code></a> is back to multiprocessing <span class="smaller">F Ramirez</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.read.html#scanpy.read" title="scanpy.read"><code class="xref py py-func docutils literal notranslate"><span class="pre">read()</span></code></a> reads compressed text files <span class="smaller">G Eraslan</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.queries.mitochondrial_genes.html#scanpy.queries.mitochondrial_genes" title="scanpy.queries.mitochondrial_genes"><code class="xref py py-func docutils literal notranslate"><span class="pre">mitochondrial_genes()</span></code></a> for querying mito genes <span class="smaller">FG Brundu</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.pp.mnn_correct.html#scanpy.external.pp.mnn_correct" title="scanpy.external.pp.mnn_correct"><code class="xref py py-func docutils literal notranslate"><span class="pre">mnn_correct()</span></code></a> for batch correction <a class="reference internal" href="../references.html#haghverdi18" id="id42"><span>[Haghverdi18]</span></a> <a class="reference internal" href="../references.html#kang18" id="id43"><span>[Kang18]</span></a></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.phate.html#scanpy.external.tl.phate" title="scanpy.external.tl.phate"><code class="xref py py-func docutils literal notranslate"><span class="pre">phate()</span></code></a> for low-dimensional embedding <a class="reference internal" href="../references.html#moon17" id="id44"><span>[Moon17]</span></a> <span class="smaller">S Gigante</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.external.tl.sandbag.html#scanpy.external.tl.sandbag" title="scanpy.external.tl.sandbag"><code class="xref py py-func docutils literal notranslate"><span class="pre">sandbag()</span></code></a>, <a class="reference internal" href="../generated/scanpy.external.tl.cyclone.html#scanpy.external.tl.cyclone" title="scanpy.external.tl.cyclone"><code class="xref py py-func docutils literal notranslate"><span class="pre">cyclone()</span></code></a> for scoring genes <a class="reference internal" href="../references.html#scialdone15" id="id45"><span>[Scialdone15]</span></a> <a class="reference internal" href="../references.html#fechtner18" id="id46"><span>[Fechtner18]</span></a></p></li>
</ul>
</section>
</section>
<section id="version-1-0">
<h2>Version 1.0<a class="headerlink" href="#version-1-0" title="Permalink to this headline"></a></h2>
<section id="id47">
<h3>1.0.0 <small>2018-03-30</small><a class="headerlink" href="#id47" title="Permalink to this headline"></a></h3>
<p class="rubric">Major updates</p>
<ul class="simple">
<li><p>Scanpy is much faster and more memory efficient: preprocess, cluster and
visualize 1.3M cells in <a class="reference external" href="https://github.com/theislab/scanpy_usage/blob/master/170522_visualizing_one_million_cells/">6h</a>, 130K cells in <a class="reference external" href="https://github.com/theislab/scanpy_usage/blob/master/170522_visualizing_one_million_cells/logfile_130K.txt">14min</a>, and 68K cells in <a class="reference external" href="https://nbviewer.jupyter.org/github/theislab/scanpy_usage/blob/master/170503_zheng17/zheng17.ipynb">3min</a> <span class="smaller">A Wolf</span></p></li>
<li><p>the API gained a preprocessing function <a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a> and a
class <a class="reference internal" href="../generated/scanpy.Neighbors.html#scanpy.Neighbors" title="scanpy.Neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">Neighbors()</span></code></a> to which all basic graph computations are
delegated <span class="smaller">A Wolf</span></p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p class="rubric">Upgrading to 1.0 isn’t fully backwards compatible in the following changes</p>
<ul class="simple">
<li><p>the graph-based tools <a class="reference internal" href="../generated/scanpy.tl.louvain.html#scanpy.tl.louvain" title="scanpy.tl.louvain"><code class="xref py py-func docutils literal notranslate"><span class="pre">louvain()</span></code></a>
<a class="reference internal" href="../generated/scanpy.tl.dpt.html#scanpy.tl.dpt" title="scanpy.tl.dpt"><code class="xref py py-func docutils literal notranslate"><span class="pre">dpt()</span></code></a> <a class="reference internal" href="../generated/scanpy.tl.draw_graph.html#scanpy.tl.draw_graph" title="scanpy.tl.draw_graph"><code class="xref py py-func docutils literal notranslate"><span class="pre">draw_graph()</span></code></a>
<a class="reference internal" href="../generated/scanpy.tl.umap.html#scanpy.tl.umap" title="scanpy.tl.umap"><code class="xref py py-func docutils literal notranslate"><span class="pre">umap()</span></code></a> <a class="reference internal" href="../generated/scanpy.tl.diffmap.html#scanpy.tl.diffmap" title="scanpy.tl.diffmap"><code class="xref py py-func docutils literal notranslate"><span class="pre">diffmap()</span></code></a>
<a class="reference internal" href="../generated/scanpy.tl.paga.html#scanpy.tl.paga" title="scanpy.tl.paga"><code class="xref py py-func docutils literal notranslate"><span class="pre">paga()</span></code></a> require prior computation of the graph:
<code class="docutils literal notranslate"><span class="pre">sc.pp.neighbors(adata,</span> <span class="pre">n_neighbors=5);</span> <span class="pre">sc.tl.louvain(adata)</span></code> instead of
previously <code class="docutils literal notranslate"><span class="pre">sc.tl.louvain(adata,</span> <span class="pre">n_neighbors=5)</span></code></p></li>
<li><p>install <code class="docutils literal notranslate"><span class="pre">numba</span></code> via <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">install</span> <span class="pre">numba</span></code>, which replaces cython</p></li>
<li><p>the default connectivity measure (dpt will look different using default
settings) changed. setting <code class="docutils literal notranslate"><span class="pre">method='gauss'</span></code> in <code class="docutils literal notranslate"><span class="pre">sc.pp.neighbors</span></code> uses
gauss kernel connectivities and reproduces the previous behavior,
see, for instance in the example <a class="reference external" href="https://nbviewer.jupyter.org/github/theislab/scanpy_usage/blob/master/170502_paul15/paul15.ipynb">paul15</a>.</p></li>
<li><p>namings of returned annotation have changed for less bloated AnnData
objects, which means that some of the unstructured annotation of old
AnnData files is not recognized anymore</p></li>
<li><p>replace occurances of <code class="docutils literal notranslate"><span class="pre">group_by</span></code> with <code class="docutils literal notranslate"><span class="pre">groupby</span></code> (consistency with
<code class="docutils literal notranslate"><span class="pre">pandas</span></code>)</p></li>
<li><p>it is worth checking out the notebook examples to see changes, e.g.
the <a class="reference external" href="https://nbviewer.jupyter.org/github/theislab/scanpy_usage/blob/master/170505_seurat/seurat.ipynb">seurat</a> example.</p></li>
<li><p>upgrading scikit-learn from 0.18 to 0.19 changed the implementation of PCA,
some results might therefore look slightly different</p></li>
</ul>
</div>
<p class="rubric">Further updates</p>
<ul class="simple">
<li><p>UMAP <a class="reference internal" href="../references.html#mcinnes18" id="id49"><span>[McInnes18]</span></a> can serve as a first visualization of the data just as tSNE,
in contrast to tSNE, UMAP directly embeds the single-cell graph and is faster;
UMAP is also used for measuring connectivities and computing neighbors,
see <a class="reference internal" href="../generated/scanpy.pp.neighbors.html#scanpy.pp.neighbors" title="scanpy.pp.neighbors"><code class="xref py py-func docutils literal notranslate"><span class="pre">neighbors()</span></code></a> <span class="smaller">A Wolf</span></p></li>
<li><p>graph abstraction: AGA is renamed to <a class="reference external" href="https://github.com/theislab/paga">PAGA</a>: <a class="reference internal" href="../generated/scanpy.tl.paga.html#scanpy.tl.paga" title="scanpy.tl.paga"><code class="xref py py-func docutils literal notranslate"><span class="pre">paga()</span></code></a>; now,
it only measures connectivities between partitions of the single-cell graph,
pseudotime and clustering need to be computed separately via
<a class="reference internal" href="../generated/scanpy.tl.louvain.html#scanpy.tl.louvain" title="scanpy.tl.louvain"><code class="xref py py-func docutils literal notranslate"><span class="pre">louvain()</span></code></a> and <a class="reference internal" href="../generated/scanpy.tl.dpt.html#scanpy.tl.dpt" title="scanpy.tl.dpt"><code class="xref py py-func docutils literal notranslate"><span class="pre">dpt()</span></code></a>, the
connectivity measure has been improved <span class="smaller">A Wolf</span></p></li>
<li><p>logistic regression for finding marker genes
<a class="reference internal" href="../generated/scanpy.tl.rank_genes_groups.html#scanpy.tl.rank_genes_groups" title="scanpy.tl.rank_genes_groups"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups()</span></code></a> with parameter <code class="docutils literal notranslate"><span class="pre">method='logreg'</span></code> <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.tl.louvain.html#scanpy.tl.louvain" title="scanpy.tl.louvain"><code class="xref py py-func docutils literal notranslate"><span class="pre">louvain()</span></code></a> provides a better implementation for
reclustering via <code class="docutils literal notranslate"><span class="pre">restrict_to</span></code> <span class="smaller">A Wolf</span></p></li>
<li><p>scanpy no longer modifies rcParams upon import, call
<code class="docutils literal notranslate"><span class="pre">settings.set_figure_params</span></code> to set the ‘scanpy style’ <span class="smaller">A Wolf</span></p></li>
<li><p>default cache directory is <code class="docutils literal notranslate"><span class="pre">./cache/</span></code>, set <code class="docutils literal notranslate"><span class="pre">settings.cachedir</span></code> to change
this; nested directories in this are avoided <span class="smaller">A Wolf</span></p></li>
<li><p>show edges in scatter plots based on graph visualization
<a class="reference internal" href="../generated/scanpy.tl.draw_graph.html#scanpy.tl.draw_graph" title="scanpy.tl.draw_graph"><code class="xref py py-func docutils literal notranslate"><span class="pre">draw_graph()</span></code></a> and <a class="reference internal" href="../generated/scanpy.tl.umap.html#scanpy.tl.umap" title="scanpy.tl.umap"><code class="xref py py-func docutils literal notranslate"><span class="pre">umap()</span></code></a> by passing <code class="docutils literal notranslate"><span class="pre">edges=True</span></code> <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pp.downsample_counts.html#scanpy.pp.downsample_counts" title="scanpy.pp.downsample_counts"><code class="xref py py-func docutils literal notranslate"><span class="pre">downsample_counts()</span></code></a> for downsampling counts <span class="smaller">MD Luecken</span></p></li>
<li><p>default <code class="docutils literal notranslate"><span class="pre">'louvain_groups'</span></code> are called <code class="docutils literal notranslate"><span class="pre">'louvain'</span></code> <span class="smaller">A Wolf</span></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">'X_diffmap'</span></code> contains the zero component, plotting remains unchanged <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
</section>
<section id="version-0-4">
<h2>Version 0.4<a class="headerlink" href="#version-0-4" title="Permalink to this headline"></a></h2>
<section id="id51">
<h3>0.4.3 <small>2018-02-09</small><a class="headerlink" href="#id51" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.pl.clustermap.html#scanpy.pl.clustermap" title="scanpy.pl.clustermap"><code class="xref py py-func docutils literal notranslate"><span class="pre">clustermap()</span></code></a>: heatmap from hierarchical clustering,
based on <a class="reference external" href="https://seaborn.pydata.org/generated/seaborn.clustermap.html#seaborn.clustermap" title="(in seaborn v0.11.2)"><code class="xref py py-func docutils literal notranslate"><span class="pre">seaborn.clustermap()</span></code></a> <a class="reference internal" href="../references.html#waskom16" id="id52"><span>[Waskom16]</span></a> <span class="smaller">A Wolf</span></p></li>
<li><p>only return <a class="reference external" href="https://matplotlib.org/stable/api/axes_api.html#matplotlib.axes.Axes" title="(in Matplotlib v3.5.1)"><code class="xref py py-class docutils literal notranslate"><span class="pre">matplotlib.axes.Axes</span></code></a> in plotting functions of <code class="docutils literal notranslate"><span class="pre">sc.pl</span></code>
when <code class="docutils literal notranslate"><span class="pre">show=False</span></code>, otherwise <code class="docutils literal notranslate"><span class="pre">None</span></code> <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
<section id="id53">
<h3>0.4.2 <small>2018-01-07</small><a class="headerlink" href="#id53" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>amendments in <a class="reference external" href="https://github.com/theislab/paga">PAGA</a> and its plotting functions <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
<section id="id55">
<h3>0.4.0 <small>2017-12-23</small><a class="headerlink" href="#id55" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>export to <a class="reference external" href="https://github.com/AllonKleinLab/SPRING/">SPRING</a> <a class="reference internal" href="../references.html#weinreb17" id="id56"><span>[Weinreb17]</span></a> for interactive visualization of data:
<a class="reference external" href="https://github.com/theislab/scanpy_usage/tree/master/171111_SPRING_export">spring tutorial</a> <span class="smaller">S Wollock</span></p></li>
</ul>
</section>
</section>
<section id="version-0-3">
<h2>Version 0.3<a class="headerlink" href="#version-0-3" title="Permalink to this headline"></a></h2>
<section id="id57">
<h3>0.3.2 <small>2017-11-29</small><a class="headerlink" href="#id57" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>finding marker genes via <a class="reference internal" href="../generated/scanpy.pl.rank_genes_groups_violin.html#scanpy.pl.rank_genes_groups_violin" title="scanpy.pl.rank_genes_groups_violin"><code class="xref py py-func docutils literal notranslate"><span class="pre">rank_genes_groups_violin()</span></code></a> improved,
see <a class="issue reference external" href="https://github.com/theislab/scanpy/issues/51">issue 51</a> <span class="smaller">F Ramirez</span></p></li>
</ul>
</section>
<section id="id58">
<h3>0.3.0 <small>2017-11-16</small><a class="headerlink" href="#id58" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.html#anndata.AnnData" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code></a> gains method <a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.concatenate.html#anndata.AnnData.concatenate" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-meth docutils literal notranslate"><span class="pre">concatenate()</span></code></a> <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference external" href="https://anndata.readthedocs.io/en/stable/anndata.AnnData.html#anndata.AnnData" title="(in anndata v0.7.7.dev1+g33006dc)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnnData</span></code></a> is available as the separate <a class="reference external" href="https://pypi.org/project/anndata/">anndata</a> package <span class="smaller">P Angerer, A Wolf</span></p></li>
<li><p>results of <a class="reference external" href="https://github.com/theislab/paga">PAGA</a> simplified <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
</section>
<section id="version-0-2">
<h2>Version 0.2<a class="headerlink" href="#version-0-2" title="Permalink to this headline"></a></h2>
<section id="id60">
<h3>0.2.9 <small>2017-10-25</small><a class="headerlink" href="#id60" title="Permalink to this headline"></a></h3>
<p class="rubric">Initial release of the new trajectory inference method <a class="reference external" href="https://github.com/theislab/paga">PAGA</a></p>
<ul class="simple">
<li><p><a class="reference internal" href="../generated/scanpy.tl.paga.html#scanpy.tl.paga" title="scanpy.tl.paga"><code class="xref py py-func docutils literal notranslate"><span class="pre">paga()</span></code></a> computes an abstracted, coarse-grained (PAGA) graph of the neighborhood graph <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.paga_compare.html#scanpy.pl.paga_compare" title="scanpy.pl.paga_compare"><code class="xref py py-func docutils literal notranslate"><span class="pre">paga_compare()</span></code></a> plot this graph next an embedding <span class="smaller">A Wolf</span></p></li>
<li><p><a class="reference internal" href="../generated/scanpy.pl.paga_path.html#scanpy.pl.paga_path" title="scanpy.pl.paga_path"><code class="xref py py-func docutils literal notranslate"><span class="pre">paga_path()</span></code></a> plots a heatmap through a node sequence in the PAGA graph <span class="smaller">A Wolf</span></p></li>
</ul>
</section>
<section id="id62">
<h3>0.2.1 <small>2017-07-24</small><a class="headerlink" href="#id62" title="Permalink to this headline"></a></h3>
<p>Scanpy includes preprocessing, visualization, clustering, pseudotime and
trajectory inference, differential expression testing and simulation of gene
regulatory networks. The implementation efficiently deals with <a class="reference external" href="https://github.com/theislab/scanpy_usage/tree/master/170522_visualizing_one_million_cells">datasets of more
than one million cells</a>. <span class="smaller">A Wolf, P Angerer</span></p>
</section>
</section>
<section id="version-0-1">
<h2>Version 0.1<a class="headerlink" href="#version-0-1" title="Permalink to this headline"></a></h2>
<section id="id63">
<h3>0.1.0 <small>2017-05-17</small><a class="headerlink" href="#id63" title="Permalink to this headline"></a></h3>
<p>Scanpy computationally outperforms and allows reproducing both the <a class="reference external" href="https://github.com/theislab/scanpy_usage/tree/master/170503_zheng17">Cell Ranger
R kit’s</a>
and most of <a class="reference external" href="https://github.com/theislab/scanpy_usage/tree/master/170505_seurat">Seurat’s</a>
clustering workflows. <span class="smaller">A Wolf, P Angerer</span></p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../ecosystem.html" class="btn btn-neutral float-left" title="Ecosystem" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../community.html" class="btn btn-neutral float-right" title="Community" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, the Scanpy development team..
      <span class="commit">Revision <code>7b22b756</code>.
      </span></p>
  </div>

   

</footer>
        </div>
      </div>
    </section>
  </div>
  

  <div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="Versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      <span class="fa fa-book"> Read the Docs</span>
      v: latest
      <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
      <dl>
        <dt>Versions</dt>
        
          <dd><a href="/en/latest/">latest</a></dd>
        
          <dd><a href="/en/stable/">stable</a></dd>
        
      </dl>
      <dl>
        <dt>Downloads</dt>
        
      </dl>
      <dl>
        
        <dt>On Read the Docs</dt>
          <dd>
            <a href="//readthedocs.com/projects/icb-scanpy/?fromdocs=icb-scanpy">Project Home</a>
          </dd>
          <dd>
            <a href="//readthedocs.com/builds/icb-scanpy/?fromdocs=icb-scanpy">Builds</a>
          </dd>
      </dl>
    </div>
  </div>
<script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>